<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Clustering and Heatmap Visualization • BS831</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><link href="../../extra.css" rel="stylesheet">
<script src="../../extra.js"></script><meta property="og:title" content="Clustering and Heatmap Visualization">
<meta property="og:description" content="BS831">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">BS831</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Markdowns
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Introduction to Genomics Analysis</li>
    <li>
      <a href="../../articles/docs/matrixOp.html">Efficient Computation in R</a>
    </li>
    <li>
      <a href="../../articles/docs/ExpressionSet.html">The ExpressionSet Data Object</a>
    </li>
    <li>
      <a href="../../articles/docs/Distributions.html">Probability Distributions</a>
    </li>
    <li>
      <a href="../../articles/docs/Mix_Normals.html">Mixture of Normals</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Preprocessing and Quality Control</li>
    <li>
      <a href="../../articles/docs/Microarray_Processing.html">Data Acquisition and ExpressionSet Basics</a>
    </li>
    <li>
      <a href="../../articles/docs/microarrayQC.html">Quality Control</a>
    </li>
    <li>
      <a href="../../articles/docs/Dimensionality_Reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../../articles/docs/Heatmaps.html">Heatmaps</a>
    </li>
    <li>
      <a href="../../articles/docs/hclust.html">Hierarchical Clustering</a>
    </li>
    <li>
      <a href="../../articles/docs/quantileNormalization.html">Quantile Normalization</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments I: Microarrays and Linear Models</li>
    <li>
      <a href="../../articles/docs/SampleSize.html">Sample Size and Statistical Significance</a>
    </li>
    <li>
      <a href="../../articles/docs/MHTcorrection.html">Multiple-Hypothesis Correction</a>
    </li>
    <li>
      <a href="../../articles/docs/Linear_Regression.html">Linear Regression</a>
    </li>
    <li>
      <a href="../../articles/docs/diffanalLM.html">Differential Analysis as Linear Regression</a>
    </li>
    <li>
      <a href="../../articles/docs/Diffanalysis.html">Gene Expression Differential Analysis with Microarrays</a>
    </li>
    <li>
      <a href="../../articles/docs/Diffanal_LimmaBC.html">Gene Expression Differential Analysis based on Limma</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments II: RNA-seq and Generalized Lineary Models</li>
    <li>
      <a href="../../articles/docs/RNAseq_ScalNorm.html">RNA-Seq Data Scaling and Normalization</a>
    </li>
    <li>
      <a href="../../articles/docs/DiffanalysisRNAseqComparison.html">RNA-seq Differential Expression Analysis with DEseq2, edgeR and limma</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments III: Differential Enrichment Analysis</li>
    <li>
      <a href="../../articles/docs/HyperEnrichment.html">HyperGeometric Enrichment</a>
    </li>
    <li>
      <a href="../../articles/docs/KS_Enrichment.html">KS-based Enrichment</a>
    </li>
    <li>
      <a href="../../articles/docs/GenesetMapping4KS.html">Mapping genesets for KS testing</a>
    </li>
    <li>
      <a href="../../articles/docs/GeneSetProjectionGSVA.html">Gene Set Projection based on KS</a>
    </li>
    <li>
      <a href="../../articles/docs/GeneSetProjectionASSIGN.html">Geneset projection based on ASSIGN</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Classification</li>
    <li>
      <a href="../../articles/docs/featureRedundancy.html">Feature Redundancy: the Good and the Bad</a>
    </li>
    <li>
      <a href="../../articles/docs/Classification_Caret.html">Classification Using Caret</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Clustering</li>
    <li>
      <a href="../../articles/docs/Clustering.html">Clustering and Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/montilab/BS831">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Clustering_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Clustering and Heatmap Visualization</h1>
                        <h4 class="author">Stefano Monti</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/montilab/BS831/blob/master/vignettes/docs/Clustering.Rmd"><code>vignettes/docs/Clustering.Rmd</code></a></small>
      <div class="hidden name"><code>Clustering.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/montilab/BS831">BS831</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">Biobase</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://mclust-org.github.io/mclust/">mclust</a></span><span class="op">)</span>
<span class="co">#source("http://bioconductor.org/biocLite.R")</span>
<span class="co">#biocLite("ConsensusClusterPlus")</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ConsensusClusterPlus</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">cba</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></code></pre></div>
<p>In this module, we illustrate the use of Hierarchical Clustering, Consensus Clustering, and Model Based Clustering (or Probabilistic Clustering) based on Finite Mixtures.</p>
<p>Notice that in this case, all three metods correctly partition the data, and Consensus Clustering and Probabilistic Clustering select the ‘correct’ number of clusters. We put ‘correct’ between quotes, since the known disease state partition is not necessarily the data-driven dominant partition (see <a href="http://montilab.bumc.bu.edu/~montilab/bibliography/MachineLearning2003_052_0091_ConsensusClustering.pdf">Monti et al., Machine Learning 2003</a>, Section 4.2.2, and note) for a discussion of this point).</p>
<div id="loading-the-data" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-the-data" class="anchor"></a>Loading the Data</h3>
<p>We start by uploading the necessary data (a Breast Cancer dataset), and restricting the dataset to the 3 largest sample groups.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">AEDAT.collapsed.mad4k</span><span class="op">)</span>

<span class="va">eset</span> <span class="op">&lt;-</span> <span class="va">AEDAT.collapsed.mad4k</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">eset</span><span class="op">$</span><span class="va">Characteristics.DiseaseState</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##    BRCA1-associated      non-basal-like              normal sporadic basal-like 
##                   2                  20                   7                  18</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## restrict to 3 classes</span>
<span class="va">samples.keep</span> <span class="op">&lt;-</span> <span class="va">eset</span><span class="op">$</span><span class="va">Characteristics.DiseaseState</span> <span class="op">%in%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"non-basal-like"</span>,<span class="st">"sporadic basal-like"</span>,<span class="st">"normal"</span><span class="op">)</span>
<span class="va">cancerSet</span> <span class="op">&lt;-</span> <span class="va">eset</span><span class="op">[</span>,<span class="va">samples.keep</span><span class="op">]</span>
<span class="va">pheno</span> <span class="op">&lt;-</span> <span class="va">cancerSet</span><span class="op">$</span><span class="va">Characteristics.DiseaseState</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span></code></pre></div>
<pre><code>## pheno
##      non-basal-like              normal sporadic basal-like 
##                  20                   7                  18</code></pre>
</div>
<div id="hierarchical-clustering-and-heatmap" class="section level3">
<h3 class="hasAnchor">
<a href="#hierarchical-clustering-and-heatmap" class="anchor"></a>Hierarchical Clustering and Heatmap</h3>
<p>We first apply simple hierarchical clustering to the 4K-gene dataset. As it is shown below, the clustering results already perfectly recapitulate the known stratification.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## hclustering + heatmap</span>
<span class="va">assayData</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span><span class="op">(</span><span class="va">cancerSet</span><span class="op">)</span>

<span class="co">## use Eculidean distance for columns/samples</span>
<span class="co">## use ward as agglomeration rule</span>
<span class="va">hc01.col</span> <span class="op">&lt;-</span> <span class="fu">hcopt</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">assayData</span><span class="op">)</span><span class="op">)</span>,method<span class="op">=</span><span class="st">"ward.D"</span><span class="op">)</span>

<span class="co">## use 1-correlation as distance for for rows/genes</span>
<span class="co">## use ward as agglomeration rule</span>
<span class="va">hc01.row</span> <span class="op">&lt;-</span> <span class="fu">hcopt</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span><span class="op">(</span><span class="fl">1</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">assayData</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,method<span class="op">=</span><span class="st">"ward.D"</span><span class="op">)</span>

<span class="co">## making heatmap</span>
<span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">cancerSet</span><span class="op">$</span><span class="va">Characteristics.DiseaseState</span><span class="op">)</span><span class="op">)</span> 
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">assayData</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">annot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DiseaseState"</span><span class="op">)</span>

<span class="va">annotCol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>DiseaseState <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"green"</span>, <span class="st">"orange"</span>, <span class="st">"purple"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">annotCol</span><span class="op">$</span><span class="va">DiseaseState</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">annot</span><span class="op">$</span><span class="va">DiseaseState</span><span class="op">)</span>
                 
<span class="va">heatmaptitle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Breast Cancer Dataset: "</span>, <span class="st">"top MAD-filtered 4k genes"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="va">assayData</span>,
         title<span class="op">=</span><span class="va">heatmaptitle</span>,
         color<span class="op">=</span><span class="fu">colGradient</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"blue"</span>,<span class="st">"white"</span>,<span class="st">"red"</span><span class="op">)</span>,length<span class="op">=</span><span class="fl">15</span><span class="op">)</span>,
         annotation_col <span class="op">=</span> <span class="va">annot</span>,
         annotation_colors <span class="op">=</span> <span class="va">annotCol</span>,
         cluster_rows<span class="op">=</span><span class="va">hc01.row</span>,
         cluster_cols<span class="op">=</span><span class="va">hc01.col</span>,
         show_rownames <span class="op">=</span> <span class="cn">FALSE</span>,
         show_colnames <span class="op">=</span> <span class="cn">FALSE</span>,
         scale <span class="op">=</span> <span class="st">"row"</span><span class="op">)</span></code></pre></div>
<p><img src="Clustering_files/figure-html/hclust-1.png" width="700"></p>
</div>
<div id="consensus-clustering-and-heatmap" class="section level3">
<h3 class="hasAnchor">
<a href="#consensus-clustering-and-heatmap" class="anchor"></a>Consensus Clustering and Heatmap</h3>
<p>We next apply Consensus Clustering, to try to determine, in a data-driven fashion, the number of clusters.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## next, run consensus clustering</span>
<span class="va">CCout</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ConsensusClusterPlus/man/ConsensusClusterPlus.html">ConsensusClusterPlus</a></span><span class="op">(</span><span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span><span class="op">(</span><span class="va">cancerSet</span><span class="op">)</span>,maxK<span class="op">=</span><span class="fl">6</span>,reps<span class="op">=</span><span class="fl">50</span>,pItem<span class="op">=</span><span class="fl">0.8</span>,pFeature<span class="op">=</span><span class="fl">1</span>,
                              innerLinkage<span class="op">=</span><span class="st">"ward.D"</span>, finalLinkage<span class="op">=</span><span class="st">"ward.D"</span>,
                              title<span class="op">=</span><span class="st">"cc"</span>,clusterAlg<span class="op">=</span><span class="st">"hc"</span>,distance<span class="op">=</span><span class="st">"euclidean"</span>,seed<span class="op">=</span><span class="fl">1262118388.71279</span>,
                              plot<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<p><img src="Clustering_files/figure-html/consensus.clustering-1.png" width="700"><img src="Clustering_files/figure-html/consensus.clustering-2.png" width="700"><img src="Clustering_files/figure-html/consensus.clustering-3.png" width="700"><img src="Clustering_files/figure-html/consensus.clustering-4.png" width="700"><img src="Clustering_files/figure-html/consensus.clustering-5.png" width="700"><img src="Clustering_files/figure-html/consensus.clustering-6.png" width="700"><img src="Clustering_files/figure-html/consensus.clustering-7.png" width="700"><img src="Clustering_files/figure-html/consensus.clustering-8.png" width="700"><img src="Clustering_files/figure-html/consensus.clustering-9.png" width="700"></p>
<p>Based on the inspection of the consensus clustering plots, we select 3 as the ‘optimal’ number of clusters.</p>
<p>Consensus Clustering partitions the samples according to the known disease state.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## notice the use of a variable (rather then using 3 directly in the following commands)</span>
<span class="va">nc</span> <span class="op">&lt;-</span> <span class="fl">3</span>

<span class="co">## remake heatmap, include both subtype and cluster assignments for visual comparison</span>
<span class="va">annot1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">annot</span>,cluster<span class="op">=</span><span class="va">CCout</span><span class="op">[[</span><span class="va">nc</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">consensusClass</span><span class="op">)</span>
<span class="va">annotCol</span><span class="op">$</span><span class="va">cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html">rainbow</a></span><span class="op">(</span>n<span class="op">=</span><span class="va">nc</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">annotCol</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">annot1</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span>

<span class="co">## use cluster tree from consensus clustering for column ordering in heatmap</span>
<span class="va">clust.col</span> <span class="op">&lt;-</span> <span class="va">CCout</span><span class="op">[[</span><span class="va">nc</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">consensusTree</span>
<span class="co">## determine row ordering based on de-novo clustering</span>
<span class="va">clust.row</span> <span class="op">&lt;-</span> <span class="fu">hcopt</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span><span class="op">(</span><span class="fl">1</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span><span class="op">(</span><span class="va">cancerSet</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,method<span class="op">=</span><span class="st">"ward.D"</span><span class="op">)</span>

<span class="co">##featureNames(cancerSet) &lt;- fData(cancerSet)$hgnc_symbol</span>
<span class="va">heatmaptitle</span> <span class="op">&lt;-</span> <span class="st">"Heatmap consensus clustering assignment vs. subtype\n top 4k z-score normalized"</span>
<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span><span class="op">(</span><span class="va">cancerSet</span><span class="op">)</span>,
         title<span class="op">=</span><span class="va">heatmaptitle</span>,
         color<span class="op">=</span><span class="fu">colGradient</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"blue"</span>,<span class="st">"white"</span>,<span class="st">"red"</span><span class="op">)</span>,length<span class="op">=</span><span class="fl">15</span><span class="op">)</span>,
         annotation_col <span class="op">=</span> <span class="va">annot</span>,
         annotation_colors <span class="op">=</span> <span class="va">annotCol</span>,
         cluster_rows<span class="op">=</span><span class="va">clust.row</span>,
         cluster_cols<span class="op">=</span><span class="va">clust.col</span>,
         show_rownames <span class="op">=</span> <span class="cn">FALSE</span>,
         show_colnames <span class="op">=</span> <span class="cn">FALSE</span>,
         scale <span class="op">=</span> <span class="st">"row"</span><span class="op">)</span></code></pre></div>
<p><img src="Clustering_files/figure-html/cc.heatmap-1.png" width="700"></p>
</div>
<div id="probabilistic-clustering-finite-mixtures" class="section level3">
<h3 class="hasAnchor">
<a href="#probabilistic-clustering-finite-mixtures" class="anchor"></a>Probabilistic Clustering (Finite Mixtures)</h3>
<p>Finally, we apply Probabilistic Clustering as implemented in <code>Mclust</code>. We score models with numbers of clusters in the 1-4 range.</p>
<p>Probabilisti Clustering also selects the ‘correct’ number of clusters (3) and partitions samples according to the known disease state.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mclust-org.github.io/mclust/reference/Mclust.html">Mclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span><span class="op">(</span><span class="va">cancerSet</span><span class="op">)</span><span class="op">)</span>,G<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">MC</span><span class="op">)</span></code></pre></div>
<pre><code>## ---------------------------------------------------- 
## Gaussian finite mixture model fitted by EM algorithm 
## ---------------------------------------------------- 
## 
## Mclust VEI (diagonal, equal shape) model with 3 components: 
## 
##  log-likelihood  n    df       BIC       ICL
##       -200111.4 45 16004 -461144.6 -461144.6
## 
## Clustering table:
##  1  2  3 
## 18 20  7</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## show cluster assignments (and disease state)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">MC</span><span class="op">$</span><span class="va">classification</span>,<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span><span class="op">(</span><span class="va">cancerSet</span><span class="op">)</span><span class="op">$</span><span class="va">Characteristics.DiseaseState</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##          MC.classification pData.cancerSet..Characteristics.DiseaseState
## GSM85480                 1                           sporadic basal-like
## GSM85490                 1                           sporadic basal-like
## GSM85500                 2                                non-basal-like
## GSM85484                 1                           sporadic basal-like
## GSM85478                 1                           sporadic basal-like
## GSM85516                 3                                        normal</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## contingency table</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/ftable.html">ftable</a></span><span class="op">(</span><span class="va">MC</span><span class="op">$</span><span class="va">classification</span>,<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span><span class="op">(</span><span class="va">cancerSet</span><span class="op">)</span><span class="op">$</span><span class="va">Characteristics.DiseaseState</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##    non-basal-like normal sporadic basal-like
##                                             
## 1               0      0                  18
## 2              20      0                   0
## 3               0      7                   0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/tetomonti">Stefano Monti</a>, <a href="https://github.com/anfederico">Anthony Federico</a>, <a href="https://github.com/ericreed">Eric Reed</a>, <a href="https://github.com/lia978">Amy Li</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
