<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Geneset projection based on ASSIGN • BS831</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><link href="../../extra.css" rel="stylesheet">
<script src="../../extra.js"></script><meta property="og:title" content="Geneset projection based on ASSIGN">
<meta property="og:description" content="BS831">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">BS831</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Markdowns
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Introduction to Genomics Analysis</li>
    <li>
      <a href="../../articles/docs/matrixOp.html">Efficient Computation in R</a>
    </li>
    <li>
      <a href="../../articles/docs/ExpressionSet.html">The ExpressionSet Data Object</a>
    </li>
    <li>
      <a href="../../articles/docs/Distributions.html">Probability Distributions</a>
    </li>
    <li>
      <a href="../../articles/docs/Mix_Normals.html">Mixture of Normals</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Preprocessing and Quality Control</li>
    <li>
      <a href="../../articles/docs/Microarray_Processing.html">Data Acquisition and ExpressionSet Basics</a>
    </li>
    <li>
      <a href="../../articles/docs/microarrayQC.html">Quality Control</a>
    </li>
    <li>
      <a href="../../articles/docs/Dimensionality_Reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../../articles/docs/Heatmaps.html">Heatmaps</a>
    </li>
    <li>
      <a href="../../articles/docs/hclust.html">Hierarchical Clustering</a>
    </li>
    <li>
      <a href="../../articles/docs/quantileNormalization.html">Quantile Normalization</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Comparative Experiments I: Microarrays and Linear Models</li>
    <li>
      <a href="../../articles/docs/SampleSize.html">Sample Size and Statistical Significance</a>
    </li>
    <li>
      <a href="../../articles/docs/MHTcorrection.html">Multiple-Hypothesis Correction</a>
    </li>
    <li>
      <a href="../../articles/docs/Linear_Regression.html">Linear Regression</a>
    </li>
    <li>
      <a href="../../articles/docs/diffanalLM.html">Differential Analysis as Linear Regression</a>
    </li>
    <li>
      <a href="../../articles/docs/Diffanalysis.html">Gene Expression Differential Analysis with Microarrays</a>
    </li>
    <li>
      <a href="../../articles/docs/Diffanal_LimmaBC.html">Gene Expression Differential Analysis based on Limma</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Comparative Experiments II: RNA-seq and Generalized Lineary Models</li>
    <li>
      <a href="../../articles/docs/RNAseq_ScalNorm.html">RNA-Seq Data Scaling and Normalization</a>
    </li>
    <li>
      <a href="../../articles/docs/DiffanalysisRNAseqComparison.html">RNA-seq Differential Expression Analysis with DEseq2, edgeR and limma</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Comparative Experiments III: Differential Enrichment Analysis</li>
    <li>
      <a href="../../articles/docs/HyperEnrichment.html">HyperGeometric Enrichment</a>
    </li>
    <li>
      <a href="../../articles/docs/KS_Enrichment.html">KS-based Enrichment</a>
    </li>
    <li>
      <a href="../../articles/docs/GenesetMapping4KS.html">Mapping genesets for KS testing</a>
    </li>
    <li>
      <a href="../../articles/docs/GeneSetProjectionGSVA.html">Gene Set Projection based on KS</a>
    </li>
    <li>
      <a href="../../articles/docs/GeneSetProjectionASSIGN.html">Geneset projection based on ASSIGN</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Classification</li>
    <li>
      <a href="../../articles/docs/featureRedundancy.html">Feature Redundancy: the Good and the Bad</a>
    </li>
    <li>
      <a href="../../articles/docs/Classification_Caret.html">Classification Using Caret</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Clustering</li>
    <li>
      <a href="../../articles/docs/Clustering.html">Clustering and Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/montilab/BS831">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Geneset projection based on ASSIGN</h1>
                        <h4 data-toc-skip class="author">Stefano
Monti</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/montilab/BS831/blob/HEAD/vignettes/docs/GeneSetProjectionASSIGN.Rmd"><code>vignettes/docs/GeneSetProjectionASSIGN.Rmd</code></a></small>
      <div class="hidden name"><code>GeneSetProjectionASSIGN.Rmd</code></div>

    </div>

    
    
In this module, we illustrate the use of a method of pathway (or
geneset, or signature) projection. In particular, we show an example of
use of Adaptive Signature Selection and InteGratioN (<a href="http://www.ncbi.nlm.nih.gov/pubmed/25617415" class="external-link">ASSIGN</a>). See <a href="Rmodule_GeneSetProjectionKS.html">Rmodule_GeneSetProjectionGSVA
</a> and <a href="Rmodule_KSenrichment.html">Rmodule_KSenrichment</a>
for an example of alternative method of pathway projection based on KS
enrichment.

<p>We start by loading the ASSIGN package, available through <a href="http://www.bioconductor.org/packages/release/bioc/html/ASSIGN.html" class="external-link">Bioconductor</a>,
and by sourcing a wrapper script for its easier handling.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/montilab/BS831">BS831</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biobase" class="external-link">Biobase</a></span><span class="op">)</span>      <span class="co"># the package with the ExpressionSet definition, among others</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://compbiomed.github.io/ASSIGN/" class="external-link">ASSIGN</a></span><span class="op">)</span>       <span class="co"># the assign package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span> <span class="co"># for heatmap visualization</span></span></code></pre></div>
<p>We next load the necessary data, consisting of gene expression
profiles from the TCGA head and neck squamous carcinoma (HNSC) dataset,
and a signature corresponding to combined shRNA knockdown of TAZ and
YAP, two transcriptional effectors of the <a href="http://www.cellsignal.com/contents/science-cst-pathways-stem-cell-markers/hippo-signaling-pathway/pathways-hippo" class="external-link">hippo
pathway</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">HNSC_RNASeq_ES</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## load the GEP data</span></span>
<span><span class="va">hnsc</span> <span class="op">&lt;-</span> <span class="va">HNSC_RNASeq_ES</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">hnsc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Features  Samples </span></span>
<span><span class="co">##    20263      566</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## load the signature</span></span>
<span><span class="va">sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"BS831"</span><span class="op">)</span>, <span class="st">"yap_taz.gmt"</span><span class="op">)</span>, header<span class="op">=</span><span class="cn">F</span>,stringsAsFactors<span class="op">=</span><span class="cn">F</span>,row.names<span class="op">=</span><span class="fl">1</span> <span class="op">)</span></span>
<span><span class="co">## show the signature (one per row)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sig</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "TAZ.YAP.UP" "TAZ.YAP.DN"</span></span></code></pre>
<p>We will only project genes that go down upon YAP/TAZ knockdown for
this example.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_set</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">sig</span><span class="op">[</span><span class="st">"TAZ.YAP.DN"</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="va">gene_set</span> <span class="op">&lt;-</span> <span class="va">gene_set</span><span class="op">[</span><span class="va">gene_set</span><span class="op">!=</span><span class="st">""</span><span class="op">]</span> <span class="co"># Strip off any empty entries</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gene_set</span><span class="op">)</span><span class="op">)</span>            <span class="co"># Show the size of the gene signature</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 381</span></span></code></pre>
<p>We are now ready to run ASSIGN. We specify the output directory where
all ASSIGN output will be stored. This folder will automatically be
created if it does not already exist.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">479</span><span class="op">)</span> <span class="co"># for reproducible results</span></span>
<span><span class="va">assign_out_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"yap_taz_activated_HNSC"</span> <span class="op">)</span> </span>
<span><span class="fu">run_assign</span><span class="op">(</span>ES <span class="op">=</span> <span class="va">hnsc</span>,eSig <span class="op">=</span> <span class="va">gene_set</span>, oDir <span class="op">=</span> <span class="va">assign_out_dir</span>, iter<span class="op">=</span><span class="fl">3000</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Running with signature:                   gene_set </span></span>
<span><span class="co">## Number of genes in signature:             381 </span></span>
<span><span class="co">## Overlap between signature and ESet genes: 358 </span></span>
<span><span class="co">## Run ASSIGN with mixture modelling beta:   TRUE  ..</span></span>
<span><span class="co">## Saving output to directory:               /tmp/Rtmph0YdcH/yap_taz_activated_HNSC</span></span></code></pre>
<pre><code><span><span class="co">## Done!</span></span></code></pre>
<p>Load the generated ASSIGN output object and generate heatmaps to
visualize the signature projection</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">assign_out_dir</span>,<span class="st">"output.rda"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">plotAll</span><span class="op">(</span>eSet <span class="op">=</span> <span class="va">hnsc</span>,output.data <span class="op">=</span><span class="va">output.data</span>,title <span class="op">=</span> <span class="st">"YAP/TAZ activation"</span> ,gene_list <span class="op">=</span>  <span class="va">gene_set</span><span class="op">)</span></span></code></pre></div>
<p><img src="GeneSetProjectionASSIGN_files/figure-html/plot%20projection-1.png" width="576"><img src="GeneSetProjectionASSIGN_files/figure-html/plot%20projection-2.png" width="576"></p>
<p>Geneset projection is particularly useful when we need to determine
the <em>activity</em> of important proteins, like transcription factors
(or effectors, more in general), which often is not properly captured by
their transcript expression. In these cases, looking at the coordinated
expression of their downstream targets is often more effective.</p>
<p>Let’s take a look at the gene expression profiles for YAP and TAZ
with respect to tumor grade. Below, we show that while the expression of
neither transcripts is significantly associated with grade progression,
their activity as captured by the ASSIGN scores (i.e., by the
coordinated expression of their downstream targets) is significantly
associated in the epxected direction, with higher grades associated to
higher TAZ/YAP activity.</p>
<p>We first fetch the expression values for the two genes. Note their
specific gene symbols (YAP1 and WWTR1)</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## YAP1 is the gene symbol for YAP</span></span>
<span><span class="va">yap1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">hnsc</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">hnsc</span><span class="op">)</span><span class="op">$</span><span class="va">gene_symbol</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="st">"YAP1"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## WWTR1 is the gene symbol for TAZ</span></span>
<span><span class="va">wwtr1</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">hnsc</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">hnsc</span><span class="op">)</span><span class="op">$</span><span class="va">gene_symbol</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="st">"WWTR1"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We next extract the generated ASSIGN YAP/TAZ projection scores for
each sample.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scores_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">assign_out_dir</span>,<span class="st">"pathway_activity_testset.csv"</span><span class="op">)</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">scores_file</span>,stringsAsFactors<span class="op">=</span><span class="cn">F</span>,header<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">yap.taz.scores</span> <span class="op">&lt;-</span> <span class="va">scores</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">yap.taz.scores</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##  0.0000  0.5850  0.7084  0.6881  0.8169  0.9973</span></span></code></pre>
<p>We then make a data frame combinding the yap/taz ASSIGN scores, yap
and taz gene expression and tumor grade. We will visualize these to see
the effect of geneset projection vs single gene expression analysis</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="st">"sample_ID"</span><span class="op">=</span><span class="va">scores</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>                <span class="st">"tumor_grade"</span><span class="op">=</span><span class="va">hnsc</span><span class="op">$</span><span class="va">my_grade</span>,</span>
<span>                <span class="st">"assign_scores"</span><span class="op">=</span><span class="va">yap.taz.scores</span>,</span>
<span>                <span class="st">"yap1"</span><span class="op">=</span><span class="va">yap1</span>,</span>
<span>                <span class="st">"wwtr1"</span><span class="op">=</span><span class="va">wwtr1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="va">d</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="va">d</span><span class="op">$</span><span class="va">tumor_grade</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">NA</span>,<span class="st">"gx"</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span> <span class="co"># Remove samples with missing grade information (NA or "gx")</span></span>
<span><span class="va">d</span><span class="op">$</span><span class="va">tumor_grade</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">d</span><span class="op">$</span><span class="va">tumor_grade</span><span class="op">)</span><span class="op">)</span> <span class="co"># Remove eliminated entries</span></span></code></pre></div>
<p>Plot individual gene expression values, as well as YAP/TAZ ASSIGN
scores with respect to tumor grade</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu">grDevices</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gray75"</span>,<span class="st">"darkgreen"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## Make a gradient of colours for different tumor grade</span></span>
<span><span class="va">grade.cols</span> <span class="op">&lt;-</span> <span class="fu">f</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">d</span><span class="op">$</span><span class="va">tumor_grade</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="co">#par(mfrow=c(3,1)) #Arrange plots in single column</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="va">yap1</span><span class="op">~</span><span class="va">tumor_grade</span>,col<span class="op">=</span><span class="va">grade.cols</span>,data <span class="op">=</span> <span class="va">d</span>,main<span class="op">=</span><span class="st">"YAP gene expression \nwrt tumor grade"</span><span class="op">)</span></span></code></pre></div>
<p><img src="GeneSetProjectionASSIGN_files/figure-html/plot%20assign%20scores%20and%20gene%20expression-1.png" width="576"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="va">wwtr1</span><span class="op">~</span><span class="va">tumor_grade</span>,col<span class="op">=</span><span class="va">grade.cols</span>,data <span class="op">=</span> <span class="va">d</span>,main<span class="op">=</span><span class="st">"TAZ gene expression \nwrt tumor grade"</span><span class="op">)</span></span></code></pre></div>
<p><img src="GeneSetProjectionASSIGN_files/figure-html/plot%20assign%20scores%20and%20gene%20expression-2.png" width="576"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="va">assign_scores</span><span class="op">~</span><span class="va">tumor_grade</span>,col<span class="op">=</span><span class="va">grade.cols</span>,data<span class="op">=</span><span class="va">d</span>,main<span class="op">=</span><span class="st">"YAP/TAZ ASSIGN scores \nwrt tumor grade"</span><span class="op">)</span></span></code></pre></div>
<p><img src="GeneSetProjectionASSIGN_files/figure-html/plot%20assign%20scores%20and%20gene%20expression-3.png" width="576"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## association with TAZ/YAP "activity" highly significant</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/anova.html" class="external-link">anova</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">assign_scores</span><span class="op">~</span><span class="va">tumor_grade</span>,data<span class="op">=</span><span class="va">d</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Analysis of Variance Table</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Response: assign_scores</span></span>
<span><span class="co">##              Df  Sum Sq Mean Sq F value    Pr(&gt;F)    </span></span>
<span><span class="co">## tumor_grade   4  5.5282 1.38205  62.333 &lt; 2.2e-16 ***</span></span>
<span><span class="co">## Residuals   539 11.9508 0.02217                      </span></span>
<span><span class="co">## ---</span></span>
<span><span class="co">## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span></code></pre>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://github.com/tetomonti" class="external-link">Stefano Monti</a>, <a href="https://github.com/anfederico" class="external-link">Anthony Federico</a>, <a href="https://github.com/ericreed" class="external-link">Eric Reed</a>, <a href="https://github.com/lia978" class="external-link">Amy Li</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
