<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Heatmaps • BS831</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Heatmaps">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">BS831</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Markdowns
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Introduction to Genomics Analysis</li>
    <li>
      <a href="../../articles/docs/matrixOp.html">Efficient Computation in R</a>
    </li>
    <li>
      <a href="../../articles/docs/ExpressionSet.html">The ExpressionSet Data Object</a>
    </li>
    <li>
      <a href="../../articles/docs/Distributions.html">Probability Distributions</a>
    </li>
    <li>
      <a href="../../articles/docs/Mix_Normals.html">Mixture of Normals</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Preprocessing and Quality Control</li>
    <li>
      <a href="../../articles/docs/Microarray_Processing.html">Data Acquisition and ExpressionSet Basics</a>
    </li>
    <li>
      <a href="../../articles/docs/microarrayQC.html">Quality Control</a>
    </li>
    <li>
      <a href="../../articles/docs/Dimensionality_Reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../../articles/docs/Heatmaps.html">Heatmaps</a>
    </li>
    <li>
      <a href="../../articles/docs/hclust.html">Hierarchical Clustering</a>
    </li>
    <li>
      <a href="../../articles/docs/quantileNormalization.html">Quantile Normalization</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments I: Microarrays and Linear Models</li>
    <li>
      <a href="../../articles/docs/SampleSize.html">Sample Size and Statistical Significance</a>
    </li>
    <li>
      <a href="../../articles/docs/MHTcorrection.html">Multiple-Hypothesis Correction</a>
    </li>
    <li>
      <a href="../../articles/docs/Linear_Regression.html">Linear Regression</a>
    </li>
    <li>
      <a href="../../articles/docs/diffanalLM.html">Differential Analysis as Linear Regression</a>
    </li>
    <li>
      <a href="../../articles/docs/Diffanalysis.html">Gene Expression Differential Analysis with Microarrays</a>
    </li>
    <li>
      <a href="../../articles/docs/Diffanal_LimmaBC.html">Gene Expression Differential Analysis based on Limma</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments II: RNA-seq and Generalized Lineary Models</li>
    <li>
      <a href="../../articles/docs/RNAseq_ScalNorm.html">RNA-Seq Data Scaling and Normalization</a>
    </li>
    <li>
      <a href="../../articles/docs/DiffanalysisRNAseqComparison.html">RNA-seq Differential Expression Analysis with DEseq2, edgeR and limma</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments III: Differential Enrichment Analysis</li>
    <li>
      <a href="../../articles/docs/HyperEnrichment.html">HyperGeometric Enrichment</a>
    </li>
    <li>
      <a href="../../articles/docs/KS_Enrichment.html">KS-based Enrichment</a>
    </li>
    <li>
      <a href="../../articles/docs/GeneSetProjectionGSVA.html">Gene Set Projection based on KS</a>
    </li>
    <li>
      <a href="../../articles/docs/GeneSetProjectionASSIGN.html">Geneset projection based on ASSIGN</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Classification</li>
    <li>
      <a href="../../articles/docs/featureRedundancy.html">Feature Redundancy: the Good and the Bad</a>
    </li>
    <li>
      <a href="../../articles/docs/Classification_Caret.html">Classification Using Caret</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Clustering</li>
    <li>
      <a href="../../articles/docs/GenesetMapping4KS.html">Mapping genesets for KS testing</a>
    </li>
    <li>
      <a href="../../articles/docs/Clustering.html">Clustering and Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/montilab/BS831">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Heatmaps</h1>
                        <h4 class="author">Stefano Monti, Amy Li, Eric Reed</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/montilab/BS831/blob/master/vignettes/docs/Heatmaps.Rmd"><code>vignettes/docs/Heatmaps.Rmd</code></a></small>
      <div class="hidden name"><code>Heatmaps.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(BS831)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(cba)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(heatmap.plus)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(pheatmap)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(Biobase)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(ggdendro)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(scales)</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(gtable)</span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(gridExtra)</span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ComplexHeatmap)</span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(circlize)</span></code></pre></div>
<div id="visualization-of-a-gene-expression-dataset-the-drug-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization-of-a-gene-expression-dataset-the-drug-matrix" class="anchor"></a>Visualization of a Gene Expression Dataset: the Drug Matrix</h2>
<p>We will use the DrugMatrix subset previously described for our examples. This data corresponds to chemical perturbation experiments, whereby rats are exposed to different chemicals, and their liver’s RNA is profiled. The chemicals are annotated both in terms of their carcinogenicity and their genotoxicity, and we will use these two phenotypes to color-code the samples in the visualized heatmaps.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(dm10)</span>
<span id="cb2-2"><a href="#cb2-2"></a>DM &lt;-<span class="st"> </span><span class="kw">variationFilter</span>(dm10, <span class="dt">ngenes=</span><span class="dv">500</span>,<span class="dt">do.plot=</span><span class="ot">FALSE</span>)</span></code></pre></div>
<pre><code>## Variation filtering based on mad .. done.
## Selecting top 500 by mad .. done, 500 genes selected.</code></pre>
</div>
<div id="the-function-heatmap-plus" class="section level2">
<h2 class="hasAnchor">
<a href="#the-function-heatmap-plus" class="anchor"></a>The Function <code>heatmap.plus</code>
</h2>
<p>We will begin by using <code>heatmap.plus</code> to visualize the data. Let us first define a simple function to create a color gradient to be used for coloring the gene expression heatmaps (this function is already defined in the <code>CBMRtools</code> package).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>colGradient &lt;-<span class="st"> </span><span class="cf">function</span>( cols, length, <span class="dt">cmax=</span><span class="dv">255</span> )</span>
<span id="cb4-2"><a href="#cb4-2"></a>{</span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="co">## e.g., to create a white-to-red gradient with 10 levels</span></span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="co">##</span></span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="co">##   col.gradient(cols=c('white','red'),length=10)</span></span>
<span id="cb4-6"><a href="#cb4-6"></a>  <span class="co">##</span></span>
<span id="cb4-7"><a href="#cb4-7"></a>  <span class="co">## or, to create a blue-to-white-to-red gradients with 9 colors (4 blue's, white, 4 red's)</span></span>
<span id="cb4-8"><a href="#cb4-8"></a>  <span class="co">##</span></span>
<span id="cb4-9"><a href="#cb4-9"></a>  <span class="co">##   col.gradient(cols=c('blue','white','red'),length=9)</span></span>
<span id="cb4-10"><a href="#cb4-10"></a>  <span class="co">##</span></span>
<span id="cb4-11"><a href="#cb4-11"></a>  ramp &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRamp</a></span>(cols)</span>
<span id="cb4-12"><a href="#cb4-12"></a>  <span class="kw"><a href="https://rdrr.io/r/grDevices/rgb.html">rgb</a></span>( <span class="kw">ramp</span>(<span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">length=</span>length)), <span class="dt">max=</span>cmax )</span>
<span id="cb4-13"><a href="#cb4-13"></a>}</span></code></pre></div>
<p>Next, let us establish the color coding of the samples in terms of their carcinogenicity (NA/negative/positive) and genotoxicity (NA/0/1). Notice the use of the function <code>rainbow</code> (to generate multiple colors), and the use of the function <code>match</code> to pair sample annotation to colors.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co">## color coding of the samples indicating carcinogenicity status and chemical</span></span>
<span id="cb5-2"><a href="#cb5-2"></a>chemicals &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(dm10)[,<span class="st">"CHEMICAL"</span>])</span>
<span id="cb5-3"><a href="#cb5-3"></a>chemPalette &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"white"</span>,<span class="st">"green"</span>,<span class="st">"purple"</span>) <span class="co"># we will use the same colors for carc and gtox</span></span>
<span id="cb5-4"><a href="#cb5-4"></a>CSC &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="dt">CARC=</span>chemPalette[<span class="kw"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(DM)[,<span class="st">"Carcinogen_liv"</span>],<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="ot">NA</span>,<span class="st">"NON-CARC"</span>,<span class="st">"CARCINOGEN"</span>))],</span>
<span id="cb5-5"><a href="#cb5-5"></a>             <span class="dt">GTOX=</span>chemPalette[<span class="kw"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(DM)[,<span class="st">"GenTox"</span>],<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="ot">NA</span>,<span class="dv">0</span>,<span class="dv">1</span>))],</span>
<span id="cb5-6"><a href="#cb5-6"></a>             <span class="dt">CHEM=</span><span class="kw"><a href="https://rdrr.io/r/grDevices/palettes.html">rainbow</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(chemicals))[<span class="kw"><a href="https://rdrr.io/r/base/match.html">match</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(DM)[,<span class="st">"CHEMICAL"</span>],chemicals)])</span>
<span id="cb5-7"><a href="#cb5-7"></a></span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="co">## color gradient for the expression levels (blue=down-regulated; white=neutral; red=up-regulated)</span></span>
<span id="cb5-9"><a href="#cb5-9"></a>bwrPalette &lt;-<span class="st"> </span><span class="kw">colGradient</span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"blue"</span>,<span class="st">"white"</span>,<span class="st">"red"</span>),<span class="dt">length=</span><span class="dv">13</span>)</span>
<span id="cb5-10"><a href="#cb5-10"></a></span>
<span id="cb5-11"><a href="#cb5-11"></a><span class="co">## draw the heatmap (hide row and col labels since they'd be</span></span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="co">## unreadable. Also don't sort rows and columns by hierarchical</span></span>
<span id="cb5-13"><a href="#cb5-13"></a><span class="co">## clustering, by setting Rowv and Colv to 'NA')</span></span>
<span id="cb5-14"><a href="#cb5-14"></a><span class="kw"><a href="https://rdrr.io/pkg/heatmap.plus/man/heatmap.plus.package.html">heatmap.plus</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(DM),<span class="dt">Rowv=</span><span class="ot">NA</span>,<span class="dt">Colv=</span><span class="ot">NA</span>,<span class="dt">col=</span>bwrPalette,<span class="dt">ColSideColors=</span>CSC,<span class="dt">labCol=</span><span class="ot">NA</span>,<span class="dt">labRow=</span><span class="ot">NA</span>)</span></code></pre></div>
<p><img src="Heatmaps_files/figure-html/heatplus1-1.png" width="700"></p>
<p>We now plot the heatmap with the rows and columns sorted by hierarchical clustering (see <code>Rmodule_hclust.Rmd</code>), by <em>not</em> setting the parameters <code>Rowv</code> and <code>Colv</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="kw"><a href="https://rdrr.io/pkg/heatmap.plus/man/heatmap.plus.package.html">heatmap.plus</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(DM),<span class="dt">col=</span>bwrPalette,<span class="dt">ColSideColors=</span>CSC,<span class="dt">labCol=</span><span class="ot">NA</span>,<span class="dt">labRow=</span><span class="ot">NA</span>)</span></code></pre></div>
<p><img src="Heatmaps_files/figure-html/heatplus2-1.png" width="700"></p>
<p>Notice that the color-coding of the expression matrix is somewhat saturated, and does not allow for the highlighting of differential expression among samples. The subsequent visualization functions have a more sophisticated coding of the expression ranges, and will allow for better visualization.</p>
</div>
<div id="the-function-pheatmap" class="section level2">
<h2 class="hasAnchor">
<a href="#the-function-pheatmap" class="anchor"></a>The Function <code>pheatmap</code>
</h2>
<p>We next illustrate the use of the function <code>pheatmap</code> from the <code>pheatmap</code> package. This function only requires a numeric matrix as input. As with <code>heatmap.plus</code> it allows for annotation of columns and rows, but with different formatting requirements. It can perform different clustering methods on rows an columns, either by specifying parameters of the clustering method to use, or by inputting the output of free-standing clustering functions such as <code>hclust</code>. Check <code><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">?pheatmap</a></code> to explore different arguments you can set, there are a lot of them.</p>
<p>First we will format the phenotype annotation for the columns of the heatmap, as well as specify the colors for each category for each variable.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>annot &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(DM)[, <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Carcinogen_liv"</span>, <span class="st">"GenTox"</span>, <span class="st">"CHEMICAL"</span>)]</span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(annot) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"CARC"</span>, <span class="st">"GTOX"</span>, <span class="st">"CHEM"</span>)</span>
<span id="cb7-3"><a href="#cb7-3"></a>annot<span class="op">$</span>CHEM[annot<span class="op">$</span>CHEM <span class="op">==</span><span class="st"> ""</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb7-4"><a href="#cb7-4"></a>annot<span class="op">$</span>CHEM &lt;-<span class="st">  </span><span class="kw"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(annot<span class="op">$</span>CHEM, <span class="dt">levels =</span> <span class="kw"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(annot<span class="op">$</span>CHEM)[<span class="kw"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(annot<span class="op">$</span>CHEM)<span class="op">!=</span><span class="st">""</span>])</span>
<span id="cb7-5"><a href="#cb7-5"></a>annot<span class="op">$</span>GTOX &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(annot<span class="op">$</span>GTOX)</span>
<span id="cb7-6"><a href="#cb7-6"></a></span>
<span id="cb7-7"><a href="#cb7-7"></a>annotCol &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(</span>
<span id="cb7-8"><a href="#cb7-8"></a>  <span class="dt">CARC =</span> chemPalette[<span class="op">-</span><span class="dv">1</span>],</span>
<span id="cb7-9"><a href="#cb7-9"></a>  <span class="dt">GTOX =</span> chemPalette[<span class="op">-</span><span class="dv">1</span>],</span>
<span id="cb7-10"><a href="#cb7-10"></a>  <span class="dt">CHEM =</span> <span class="kw"><a href="https://rdrr.io/r/grDevices/palettes.html">rainbow</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(annot<span class="op">$</span>CHEM)))</span>
<span id="cb7-11"><a href="#cb7-11"></a>)</span>
<span id="cb7-12"><a href="#cb7-12"></a><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(annotCol<span class="op">$</span>CARC) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"NON-CARC"</span>,<span class="st">"CARCINOGEN"</span>)</span>
<span id="cb7-13"><a href="#cb7-13"></a><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(annotCol<span class="op">$</span>GTOX) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"0"</span>,<span class="st">"1"</span>)</span>
<span id="cb7-14"><a href="#cb7-14"></a><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(annotCol<span class="op">$</span>CHEM) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(annot<span class="op">$</span>CHEM)</span></code></pre></div>
<p>Next, we will create the heatmap, given the expression matrix from DrugMatrix. Note that the argument <code>scale = "row"</code>, specifies that we want to scale each row (or gene) to mean zero, and let the colors denote the number of standard deviations from the mean.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="kw"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(DM),</span>
<span id="cb8-2"><a href="#cb8-2"></a>        <span class="dt">color=</span>bwrPalette, </span>
<span id="cb8-3"><a href="#cb8-3"></a>        <span class="dt">annotation_col =</span> annot,</span>
<span id="cb8-4"><a href="#cb8-4"></a>        <span class="dt">annotation_colors =</span> annotCol,</span>
<span id="cb8-5"><a href="#cb8-5"></a>        <span class="dt">show_rownames =</span> <span class="ot">FALSE</span>,</span>
<span id="cb8-6"><a href="#cb8-6"></a>        <span class="dt">show_colnames =</span> <span class="ot">FALSE</span>,</span>
<span id="cb8-7"><a href="#cb8-7"></a>        <span class="dt">scale =</span> <span class="st">"row"</span>)</span></code></pre></div>
<p><img src="Heatmaps_files/figure-html/pheatplusrun-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="the-function-cbmrtoolsggheat-continuous" class="section level2">
<h2 class="hasAnchor">
<a href="#the-function-cbmrtoolsggheat-continuous" class="anchor"></a>The Function <code>CBMRtools::ggheat.continuous</code>
</h2>
<p>We next illustrate the use of a custom heatmap function that uses ggplot2 and is part of our package <code>CBMRtools</code>. This heatmap supports multiple column phenotype labeling, custom heatmap color gradient, and column clustering within user-specified sample groups.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="co">#simplify chemical carcinogen status label for plotting purposes</span></span>
<span id="cb9-2"><a href="#cb9-2"></a>DM<span class="op">$</span>CARC &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(DM<span class="op">$</span>Carcinogen_liv)</span>
<span id="cb9-3"><a href="#cb9-3"></a>DM<span class="op">$</span>CARC[<span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(DM<span class="op">$</span>CARC)] &lt;-<span class="st"> "NA"</span></span>
<span id="cb9-4"><a href="#cb9-4"></a></span>
<span id="cb9-5"><a href="#cb9-5"></a>DM<span class="op">$</span>GTOX &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(DM<span class="op">$</span>GenTox)</span>
<span id="cb9-6"><a href="#cb9-6"></a>DM<span class="op">$</span>GTOX[<span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(DM<span class="op">$</span>GTOX)] &lt;-<span class="st"> "NA"</span></span>
<span id="cb9-7"><a href="#cb9-7"></a></span>
<span id="cb9-8"><a href="#cb9-8"></a><span class="co">#specify column and row clustering</span></span>
<span id="cb9-9"><a href="#cb9-9"></a><span class="co">#clust_eset is a custom clustering function we created for standard gene expression hierarchical clustering</span></span>
<span id="cb9-10"><a href="#cb9-10"></a><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(clust_eset)</span></code></pre></div>
<pre><code>## function(eset){
##  mat&lt;-Biobase::exprs(eset)
## 
##  #column clustering
##  #using euclidean distance, ward.D agglomeration
##  distC &lt;- function(x) dist(t(x), method="euclidean")
##  dist_c&lt;-distC(mat)
##  hc&lt;-hcopt(dist_c, method = "ward.D")
## 
##  #row clustering
##  #using 1-cor as distance, ward.D agglomeration
##  distR &lt;- function(x) stats::as.dist(1- cor(t(x)))
##  dist_r&lt;-distR(mat)
##  hr&lt;-hcopt(dist_r, method = "ward.D")
## 
##  return(list(hc = hc, hr = hr))
## }
## &lt;environment: namespace:BS831&gt;</code></pre>
<p>Below, we use hierarchical clustering to group samples and genes according to their similarity. See the module <code>Rmodule_hclust.Rmd</code> for more details on the clustering method and the R function implementing it, <code>hclust</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>hc &lt;-<span class="st"> </span><span class="kw">clust_eset</span>(DM)</span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(hc) <span class="co"># show the 'innards' of the returned object</span></span></code></pre></div>
<pre><code>## $hc
## 
## Call:
## hclust(d = d, method = method, members = members)
## 
## Cluster method   : ward.D 
## Distance         : euclidean 
## Number of objects: 338 
## 
## 
## $hr
## 
## Call:
## hclust(d = d, method = method, members = members)
## 
## Cluster method   : ward.D 
## Number of objects: 500</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co"># Use column (i.e., sample) clusters as one of the phenotype labels to plot</span></span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(DM)<span class="op">$</span>hclust.groups &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="kw"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(hc<span class="op">$</span>hc, <span class="dt">k =</span> <span class="dv">4</span>))</span>
<span id="cb13-3"><a href="#cb13-3"></a></span>
<span id="cb13-4"><a href="#cb13-4"></a><span class="co"># Scale expression by row (for visualization purpose only), hence do</span></span>
<span id="cb13-5"><a href="#cb13-5"></a><span class="co"># this step after clustering is defined on pre-scaled data</span></span>
<span id="cb13-6"><a href="#cb13-6"></a><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(scale_row)</span></code></pre></div>
<pre><code>## function(eset){
##  rowz&lt;-t(apply(Biobase::exprs(eset), 1, function(z)
##          scale(z)))
##  Biobase::exprs(eset)&lt;-rowz
##  return(eset)
## }
## &lt;environment: namespace:BS831&gt;</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>dat.scaled &lt;-<span class="st"> </span><span class="kw">scale_row</span>(DM)</span>
<span id="cb15-2"><a href="#cb15-2"></a></span>
<span id="cb15-3"><a href="#cb15-3"></a>carcstatus &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"NA"</span>, <span class="st">"CARCINOGEN"</span>, <span class="st">"NON-CARC"</span>)</span>
<span id="cb15-4"><a href="#cb15-4"></a>gtoxstatus &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"NA"</span>, <span class="st">"1"</span>, <span class="st">"0"</span>)</span>
<span id="cb15-5"><a href="#cb15-5"></a>cluststatus &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(DM)<span class="op">$</span>hclust.groups)</span>
<span id="cb15-6"><a href="#cb15-6"></a></span>
<span id="cb15-7"><a href="#cb15-7"></a><span class="co"># Color legends for column labels: here we plot CARC and hclust.groups</span></span>
<span id="cb15-8"><a href="#cb15-8"></a>col_legend &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(</span>
<span id="cb15-9"><a href="#cb15-9"></a>    <span class="dt">CARC =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">col_breaks =</span> carcstatus,</span>
<span id="cb15-10"><a href="#cb15-10"></a>        <span class="dt">col_values =</span> <span class="kw"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"white"</span>, <span class="st">"purple"</span>, <span class="st">"green"</span>), to.hex),</span>
<span id="cb15-11"><a href="#cb15-11"></a>        <span class="dt">col_labels =</span> carcstatus), </span>
<span id="cb15-12"><a href="#cb15-12"></a>    <span class="dt">hclust.groups =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">col_breaks =</span> cluststatus,</span>
<span id="cb15-13"><a href="#cb15-13"></a>        <span class="dt">col_values =</span> <span class="kw"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"pink"</span>, <span class="st">"orange"</span>, <span class="st">"yellow"</span>, <span class="st">"cyan"</span>), to.hex),</span>
<span id="cb15-14"><a href="#cb15-14"></a>        <span class="dt">col_labels =</span> cluststatus)</span>
<span id="cb15-15"><a href="#cb15-15"></a>)</span>
<span id="cb15-16"><a href="#cb15-16"></a></span>
<span id="cb15-17"><a href="#cb15-17"></a><span class="co"># heatmap fill gradient</span></span>
<span id="cb15-18"><a href="#cb15-18"></a>hmcolors &lt;-<span class="st"> </span><span class="cf">function</span>(...) <span class="kw">scale_fill_gradient2</span>(<span class="dt">low =</span> <span class="st">"blue"</span>, <span class="dt">mid =</span> <span class="st">"white"</span>,</span>
<span id="cb15-19"><a href="#cb15-19"></a>       <span class="dt">high=</span><span class="st">"red"</span>, <span class="dt">midpoint=</span><span class="dv">0</span>, <span class="dt">limits=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="op">-</span><span class="dv">3</span>,<span class="dv">3</span>), <span class="dt">oob=</span>squish, ...)</span>
<span id="cb15-20"><a href="#cb15-20"></a></span>
<span id="cb15-21"><a href="#cb15-21"></a>p &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/ggheat.continuous.single.html">ggheat.continuous.single</a></span>(</span>
<span id="cb15-22"><a href="#cb15-22"></a>    <span class="dt">eset =</span> dat.scaled, </span>
<span id="cb15-23"><a href="#cb15-23"></a>    <span class="dt">hc =</span> hc<span class="op">$</span>hc, </span>
<span id="cb15-24"><a href="#cb15-24"></a>    <span class="dt">hr =</span> hc<span class="op">$</span>hr, </span>
<span id="cb15-25"><a href="#cb15-25"></a>    <span class="dt">hmcolors =</span> hmcolors,</span>
<span id="cb15-26"><a href="#cb15-26"></a>    <span class="dt">hmtitle =</span> <span class="st">"row-zscore GE"</span>,</span>
<span id="cb15-27"><a href="#cb15-27"></a>    <span class="dt">col_lab =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"CARC"</span>, <span class="st">"hclust.groups"</span>), </span>
<span id="cb15-28"><a href="#cb15-28"></a>    <span class="dt">col_legend =</span> col_legend,</span>
<span id="cb15-29"><a href="#cb15-29"></a>    <span class="dt">ylabstr =</span> <span class="st">""</span>,</span>
<span id="cb15-30"><a href="#cb15-30"></a>    <span class="dt">fout =</span> <span class="ot">NA</span>, </span>
<span id="cb15-31"><a href="#cb15-31"></a>    <span class="dt">p.heights =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1.5</span>, <span class="fl">0.5</span>, <span class="dv">5</span>),</span>
<span id="cb15-32"><a href="#cb15-32"></a>    <span class="dt">xsize =</span> <span class="dv">0</span>,</span>
<span id="cb15-33"><a href="#cb15-33"></a>    <span class="dt">ysize =</span> <span class="dv">0</span>, </span>
<span id="cb15-34"><a href="#cb15-34"></a>    <span class="dt">ysizelab =</span> <span class="dv">7</span>,</span>
<span id="cb15-35"><a href="#cb15-35"></a>    <span class="dt">xright =</span> <span class="fl">0.18</span></span>
<span id="cb15-36"><a href="#cb15-36"></a>)</span></code></pre></div>
<p><img src="Heatmaps_files/figure-html/cbmrtoolsggheat.single1-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="co"># You can see the carc/noncarc distribution within clusters is</span></span>
<span id="cb16-2"><a href="#cb16-2"></a><span class="co"># consistent with heatmap column labels</span></span>
<span id="cb16-3"><a href="#cb16-3"></a><span class="kw"><a href="https://rdrr.io/r/base/table.html">table</a></span>(DM<span class="op">$</span>CARC, DM<span class="op">$</span>hclust.groups)</span></code></pre></div>
<pre><code>##             
##                1   2   3   4
##   CARCINOGEN   7  14   0  29
##   NA         111   6  36  72
##   NON-CARC     7  15  19  22</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="co">#optional: save to image file</span></span>
<span id="cb18-2"><a href="#cb18-2"></a><span class="co">#ggsave(p, file = "example.png")</span></span></code></pre></div>
<p>We can also plot groups of heatmaps and do column clustering within each group. Here we specify the groups as (carc) <code>pos</code>, <code>neg</code>, and <code>na</code>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="co"># Color legends for column labels: here we plot CARC, hclust.groups, and GTOX</span></span>
<span id="cb19-2"><a href="#cb19-2"></a>col_legend &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(</span>
<span id="cb19-3"><a href="#cb19-3"></a>    <span class="dt">CARC =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">col_breaks =</span> carcstatus,</span>
<span id="cb19-4"><a href="#cb19-4"></a>        <span class="dt">col_values =</span> <span class="kw"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"white"</span>, <span class="st">"purple"</span>, <span class="st">"green"</span>), to.hex),</span>
<span id="cb19-5"><a href="#cb19-5"></a>        <span class="dt">col_labels =</span> carcstatus), </span>
<span id="cb19-6"><a href="#cb19-6"></a>    <span class="dt">GTOX =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">col_breaks =</span> gtoxstatus,</span>
<span id="cb19-7"><a href="#cb19-7"></a>        <span class="dt">col_values =</span> <span class="kw"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"white"</span>, <span class="st">"purple"</span>, <span class="st">"green"</span>), to.hex),</span>
<span id="cb19-8"><a href="#cb19-8"></a>        <span class="dt">col_labels =</span> gtoxstatus), </span>
<span id="cb19-9"><a href="#cb19-9"></a>    <span class="dt">hclust.groups =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">col_breaks =</span> cluststatus,</span>
<span id="cb19-10"><a href="#cb19-10"></a>        <span class="dt">col_values =</span> <span class="kw"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"pink"</span>, <span class="st">"orange"</span>, <span class="st">"yellow"</span>, <span class="st">"cyan"</span>), to.hex),</span>
<span id="cb19-11"><a href="#cb19-11"></a>        <span class="dt">col_labels =</span> cluststatus)</span>
<span id="cb19-12"><a href="#cb19-12"></a>)</span>
<span id="cb19-13"><a href="#cb19-13"></a>p &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/ggheat.continuous.single.html">ggheat.continuous.single</a></span>(</span>
<span id="cb19-14"><a href="#cb19-14"></a>    <span class="dt">eset =</span> dat.scaled, </span>
<span id="cb19-15"><a href="#cb19-15"></a>    <span class="dt">hc =</span> hc<span class="op">$</span>hc, </span>
<span id="cb19-16"><a href="#cb19-16"></a>    <span class="dt">hr =</span> hc<span class="op">$</span>hr, </span>
<span id="cb19-17"><a href="#cb19-17"></a>    <span class="dt">hmcolors =</span> hmcolors,</span>
<span id="cb19-18"><a href="#cb19-18"></a>    <span class="dt">hmtitle =</span> <span class="st">"row-zscore GE"</span>,</span>
<span id="cb19-19"><a href="#cb19-19"></a>    <span class="dt">col_lab =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"CARC"</span>, <span class="st">"GTOX"</span>, <span class="st">"hclust.groups"</span>), </span>
<span id="cb19-20"><a href="#cb19-20"></a>    <span class="dt">col_legend =</span> col_legend,</span>
<span id="cb19-21"><a href="#cb19-21"></a>    <span class="dt">ylabstr =</span> <span class="st">""</span>,</span>
<span id="cb19-22"><a href="#cb19-22"></a>    <span class="dt">fout =</span> <span class="ot">NA</span>, </span>
<span id="cb19-23"><a href="#cb19-23"></a>    <span class="dt">p.heights =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2.0</span>, <span class="fl">0.5</span>, <span class="dv">5</span>),</span>
<span id="cb19-24"><a href="#cb19-24"></a>    <span class="dt">xsize =</span> <span class="dv">0</span>,</span>
<span id="cb19-25"><a href="#cb19-25"></a>    <span class="dt">ysize =</span> <span class="dv">0</span>, </span>
<span id="cb19-26"><a href="#cb19-26"></a>    <span class="dt">ysizelab =</span> <span class="dv">7</span>,</span>
<span id="cb19-27"><a href="#cb19-27"></a>    <span class="dt">xright =</span> <span class="fl">0.18</span></span>
<span id="cb19-28"><a href="#cb19-28"></a>)</span></code></pre></div>
<p><img src="Heatmaps_files/figure-html/cbmrtoolsggheat.single2-1.png" width="700"></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>grps &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/ggheat.make.groups.html">ggheat.make.groups</a></span>(</span>
<span id="cb20-2"><a href="#cb20-2"></a>    <span class="dt">eset =</span> DM,              <span class="co"># the expression set</span></span>
<span id="cb20-3"><a href="#cb20-3"></a>    <span class="dt">labelcol =</span> <span class="st">"CARC"</span>,      <span class="co"># column name for grouping in pData(eset)</span></span>
<span id="cb20-4"><a href="#cb20-4"></a>    <span class="dt">labelvals =</span> carcstatus, <span class="co"># values to group on (e.g. factor levels of pData(eset)[, labelcol])</span></span>
<span id="cb20-5"><a href="#cb20-5"></a>    <span class="dt">clustFUN =</span> clust_eset   <span class="co"># clustering function for eset</span></span>
<span id="cb20-6"><a href="#cb20-6"></a>)</span>
<span id="cb20-7"><a href="#cb20-7"></a></span>
<span id="cb20-8"><a href="#cb20-8"></a>col_legend &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">CARC =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">col_breaks =</span> carcstatus,</span>
<span id="cb20-9"><a href="#cb20-9"></a>    <span class="dt">col_values =</span> <span class="kw"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"white"</span>, <span class="st">"purple"</span>, <span class="st">"green"</span>), to.hex),</span>
<span id="cb20-10"><a href="#cb20-10"></a>    <span class="dt">col_labels =</span> carcstatus))</span>
<span id="cb20-11"><a href="#cb20-11"></a></span>
<span id="cb20-12"><a href="#cb20-12"></a>esetlist &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(grps<span class="op">$</span>esetlist, <span class="cf">function</span>(i) <span class="kw">scale_row</span>(i))</span>
<span id="cb20-13"><a href="#cb20-13"></a></span>
<span id="cb20-14"><a href="#cb20-14"></a>p &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/ggheat.continuous.group.html">ggheat.continuous.group</a></span>(</span>
<span id="cb20-15"><a href="#cb20-15"></a>    esetlist, </span>
<span id="cb20-16"><a href="#cb20-16"></a>    grps<span class="op">$</span>hclist, </span>
<span id="cb20-17"><a href="#cb20-17"></a>    grps<span class="op">$</span>hrlist, </span>
<span id="cb20-18"><a href="#cb20-18"></a>    hmcolors,</span>
<span id="cb20-19"><a href="#cb20-19"></a>    <span class="dt">hmtitle =</span> <span class="st">"row-zscore GE"</span>,</span>
<span id="cb20-20"><a href="#cb20-20"></a>    <span class="dt">col_lab =</span> <span class="st">"CARC"</span>, </span>
<span id="cb20-21"><a href="#cb20-21"></a>    <span class="dt">col_legend =</span> col_legend,</span>
<span id="cb20-22"><a href="#cb20-22"></a>    <span class="dt">ylabstr =</span> <span class="st">""</span>,</span>
<span id="cb20-23"><a href="#cb20-23"></a>    <span class="dt">fout  =</span> <span class="ot">NA</span>, </span>
<span id="cb20-24"><a href="#cb20-24"></a>    <span class="dt">p.heights =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1.5</span>, <span class="fl">0.5</span>, <span class="dv">5</span>),</span>
<span id="cb20-25"><a href="#cb20-25"></a>    <span class="dt">xsize =</span> <span class="dv">0</span>,</span>
<span id="cb20-26"><a href="#cb20-26"></a>    <span class="dt">ysize =</span> <span class="dv">0</span>, </span>
<span id="cb20-27"><a href="#cb20-27"></a>    <span class="dt">ysizelab =</span> <span class="dv">7</span>,</span>
<span id="cb20-28"><a href="#cb20-28"></a>    <span class="dt">xleft =</span> <span class="fl">0.10</span>, </span>
<span id="cb20-29"><a href="#cb20-29"></a>    <span class="dt">xright =</span> <span class="fl">0.24</span></span>
<span id="cb20-30"><a href="#cb20-30"></a>)</span></code></pre></div>
<p><img src="Heatmaps_files/figure-html/cbmrtoolsggheat.group-1.png" width="700"></p>
</div>
<div id="the-package-complexheatmap" class="section level2">
<h2 class="hasAnchor">
<a href="#the-package-complexheatmap" class="anchor"></a>The package <code>ComplexHeatmap</code>
</h2>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="co"># Pull out expression matrix</span></span>
<span id="cb21-2"><a href="#cb21-2"></a>mat &lt;-<span class="st"> </span>Biobase<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(dat.scaled)</span>
<span id="cb21-3"><a href="#cb21-3"></a></span>
<span id="cb21-4"><a href="#cb21-4"></a><span class="co"># Take columns you want from phenotype data</span></span>
<span id="cb21-5"><a href="#cb21-5"></a>ha.t &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/HeatmapAnnotation.html">HeatmapAnnotation</a></span>(<span class="dt">CHEMICAL=</span>dat.scaled<span class="op">$</span>CARC,</span>
<span id="cb21-6"><a href="#cb21-6"></a>                          <span class="dt">GTOX=</span><span class="kw"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(dat.scaled<span class="op">$</span>GenTox))</span>
<span id="cb21-7"><a href="#cb21-7"></a><span class="kw"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html">Heatmap</a></span>(mat,</span>
<span id="cb21-8"><a href="#cb21-8"></a>        <span class="dt">name=</span><span class="st">"expression"</span>, </span>
<span id="cb21-9"><a href="#cb21-9"></a>        <span class="dt">col=</span><span class="kw"><a href="https://rdrr.io/pkg/circlize/man/colorRamp2.html">colorRamp2</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="op">-</span><span class="dv">3</span>, <span class="dv">0</span>, <span class="dv">3</span>), <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"#072448"</span>, <span class="st">"white"</span>, <span class="st">"#ff6150"</span>)),</span>
<span id="cb21-10"><a href="#cb21-10"></a>        <span class="dt">top_annotation=</span>ha.t, </span>
<span id="cb21-11"><a href="#cb21-11"></a>        <span class="dt">cluster_rows=</span><span class="ot">TRUE</span>,</span>
<span id="cb21-12"><a href="#cb21-12"></a>        <span class="dt">cluster_columns=</span><span class="ot">TRUE</span>,</span>
<span id="cb21-13"><a href="#cb21-13"></a>        <span class="dt">clustering_distance_rows=</span><span class="st">"euclidean"</span>,</span>
<span id="cb21-14"><a href="#cb21-14"></a>        <span class="dt">clustering_method_rows=</span><span class="st">"ward.D"</span>,    </span>
<span id="cb21-15"><a href="#cb21-15"></a>        <span class="dt">clustering_distance_columns=</span><span class="st">"euclidean"</span>,</span>
<span id="cb21-16"><a href="#cb21-16"></a>        <span class="dt">clustering_method_columns=</span><span class="st">"ward.D"</span>, </span>
<span id="cb21-17"><a href="#cb21-17"></a>        <span class="dt">column_km=</span><span class="dv">4</span>, </span>
<span id="cb21-18"><a href="#cb21-18"></a>        <span class="dt">show_parent_dend_line=</span><span class="ot">TRUE</span>,</span>
<span id="cb21-19"><a href="#cb21-19"></a>        <span class="dt">row_title=</span><span class="st">""</span>,</span>
<span id="cb21-20"><a href="#cb21-20"></a>        <span class="dt">show_column_names=</span><span class="ot">FALSE</span>,</span>
<span id="cb21-21"><a href="#cb21-21"></a>        <span class="dt">show_row_names=</span><span class="ot">FALSE</span>)</span></code></pre></div>
<p><img src="Heatmaps_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#visualization-of-a-gene-expression-dataset-the-drug-matrix">Visualization of a Gene Expression Dataset: the Drug Matrix</a></li>
      <li><a href="#the-function-heatmap-plus">The Function <code>heatmap.plus</code></a></li>
      <li><a href="#the-function-pheatmap">The Function <code>pheatmap</code></a></li>
      <li><a href="#the-function-cbmrtoolsggheat-continuous">The Function <code>CBMRtools::ggheat.continuous</code></a></li>
      <li><a href="#the-package-complexheatmap">The package <code>ComplexHeatmap</code></a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/tetomonti">Stefano Monti</a>, <a href="https://github.com/lia978">Amy Li</a>, <a href="https://github.com/ericreed">Eric Reed</a>, <a href="https://github.com/anfederico">Anthony Federico</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
