<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Quality Control • BS831</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><link href="../../extra.css" rel="stylesheet">
<script src="../../extra.js"></script><meta property="og:title" content="Quality Control">
<meta property="og:description" content="BS831">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">BS831</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Markdowns
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Introduction to Genomics Analysis</li>
    <li>
      <a href="../../articles/docs/matrixOp.html">Efficient Computation in R</a>
    </li>
    <li>
      <a href="../../articles/docs/ExpressionSet.html">The ExpressionSet Data Object</a>
    </li>
    <li>
      <a href="../../articles/docs/Distributions.html">Probability Distributions</a>
    </li>
    <li>
      <a href="../../articles/docs/Mix_Normals.html">Mixture of Normals</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Preprocessing and Quality Control</li>
    <li>
      <a href="../../articles/docs/Microarray_Processing.html">Data Acquisition and ExpressionSet Basics</a>
    </li>
    <li>
      <a href="../../articles/docs/microarrayQC.html">Quality Control</a>
    </li>
    <li>
      <a href="../../articles/docs/Dimensionality_Reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../../articles/docs/Heatmaps.html">Heatmaps</a>
    </li>
    <li>
      <a href="../../articles/docs/hclust.html">Hierarchical Clustering</a>
    </li>
    <li>
      <a href="../../articles/docs/quantileNormalization.html">Quantile Normalization</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments I: Microarrays and Linear Models</li>
    <li>
      <a href="../../articles/docs/SampleSize.html">Sample Size and Statistical Significance</a>
    </li>
    <li>
      <a href="../../articles/docs/MHTcorrection.html">Multiple-Hypothesis Correction</a>
    </li>
    <li>
      <a href="../../articles/docs/Linear_Regression.html">Linear Regression</a>
    </li>
    <li>
      <a href="../../articles/docs/diffanalLM.html">Differential Analysis as Linear Regression</a>
    </li>
    <li>
      <a href="../../articles/docs/Diffanalysis.html">Gene Expression Differential Analysis with Microarrays</a>
    </li>
    <li>
      <a href="../../articles/docs/Diffanal_LimmaBC.html">Gene Expression Differential Analysis based on Limma</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments II: RNA-seq and Generalized Lineary Models</li>
    <li>
      <a href="../../articles/docs/RNAseq_ScalNorm.html">RNA-Seq Data Scaling and Normalization</a>
    </li>
    <li>
      <a href="../../articles/docs/DiffanalysisRNAseqComparison.html">RNA-seq Differential Expression Analysis with DEseq2, edgeR and limma</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments III: Differential Enrichment Analysis</li>
    <li>
      <a href="../../articles/docs/HyperEnrichment.html">HyperGeometric Enrichment</a>
    </li>
    <li>
      <a href="../../articles/docs/KS_Enrichment.html">KS-based Enrichment</a>
    </li>
    <li>
      <a href="../../articles/docs/GeneSetProjectionGSVA.html">Gene Set Projection based on KS</a>
    </li>
    <li>
      <a href="../../articles/docs/GeneSetProjectionASSIGN.html">Geneset projection based on ASSIGN</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Classification</li>
    <li>
      <a href="../../articles/docs/featureRedundancy.html">Feature Redundancy: the Good and the Bad</a>
    </li>
    <li>
      <a href="../../articles/docs/Classification_Caret.html">Classification Using Caret</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Clustering</li>
    <li>
      <a href="../../articles/docs/GenesetMapping4KS.html">Mapping genesets for KS testing</a>
    </li>
    <li>
      <a href="../../articles/docs/Clustering.html">Clustering and Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/montilab/BS831">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Quality Control</h1>
                        <h4 class="author">Stefano Monti</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/montilab/BS831/blob/master/vignettes/docs/microarrayQC.Rmd"><code>vignettes/docs/microarrayQC.Rmd</code></a></small>
      <div class="hidden name"><code>microarrayQC.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BS831</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">Biobase</span>)</pre></body></html></div>
<p>Here we show some basic QC plots to assess the quality of the data and the nature of the distribution. For this purpose, we use a small Breast Cancer dataset we have pre-uploaded to the <code>data/</code> folder.</p>
<div id="import-breast-cancer-data" class="section level2">
<h2 class="hasAnchor">
<a href="#import-breast-cancer-data" class="anchor"></a>Import Breast Cancer Data</h2>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co">## breast cancer subtypes dataset from ArrayExpress</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">AEDAT.collapsed</span>)

<span class="no">eset</span> <span class="kw">&lt;-</span> <span class="no">AEDAT.collapsed</span>
<span class="no">emat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">eset</span>)

<span class="co">## Assign rownames to emat that are gene annotations (These were removed by mistake)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">emat</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(<span class="no">eset</span>))

<span class="co">## Table of disease states</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">eset</span>$<span class="no">Characteristics.DiseaseState</span>))</pre></body></html></div>
<pre><code>## 
##    BRCA1-associated breast cancer      non-basal-like breast cancer 
##                                 2                                20 
##                            normal sporadic basal-like breast cancer 
##                                 7                                18</code></pre>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="co">## simplify labels</span>
<span class="no">eset</span>$<span class="no">Characteristics.DiseaseState</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">" breast cancer"</span>,<span class="st">""</span>,<span class="no">eset</span>$<span class="no">Characteristics.DiseaseState</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">eset</span>$<span class="no">Characteristics.DiseaseState</span>))</pre></body></html></div>
<pre><code>## 
##    BRCA1-associated      non-basal-like              normal sporadic basal-like 
##                   2                  20                   7                  18</code></pre>
</div>
<div id="boxplots-of-gene-expression-values-by-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#boxplots-of-gene-expression-values-by-sample" class="anchor"></a>Boxplots of gene expression values by sample</h2>
<p>Here, we plot the distribution of genes within each sample as boxplots. This is a quick check of how symmetric the distributions are, and whether any ‘biases’ are apparent.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="co">## Boxplot of expression values per sample (sorted and colored by disease state)</span>
<span class="no">sampleCol</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"coral"</span>,<span class="st">"green"</span>,<span class="st">"blue"</span>,<span class="st">"purple"</span>)[<span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="no">eset</span>$<span class="no">Characteristics.DiseaseState</span>)]
<span class="no">ord</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>( <span class="no">eset</span>$<span class="no">Characteristics.DiseaseState</span> )
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">eset</span>)[,<span class="no">ord</span>]),<span class="kw">col</span><span class="kw">=</span><span class="no">sampleCol</span>[<span class="no">ord</span>],<span class="kw">pch</span><span class="kw">=</span><span class="st">"-"</span>)</pre></body></html></div>
<p><img src="microarrayQC_files/figure-html/boxplot-1.png" width="700"></p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co">## If you have a slow machine, you might want to subsample (random 1000 genes from the total)</span>
<span class="co">## here, we are also taking the opportunity to better size (cex) and orient (las) the axis labels</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">321</span>) <span class="co"># for reproducible results</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">eset</span>)[<span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">emat</span>),<span class="fl">1000</span>),<span class="no">ord</span>]),<span class="kw">col</span><span class="kw">=</span><span class="no">sampleCol</span>[<span class="no">ord</span>],<span class="kw">pch</span><span class="kw">=</span><span class="st">"-"</span>,
        <span class="kw">las</span><span class="kw">=</span><span class="fl">2</span>,<span class="kw">cex.axis</span><span class="kw">=</span><span class="fl">.75</span>)</pre></body></html></div>
<p><img src="microarrayQC_files/figure-html/boxplot-2.png" width="700"></p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="co">## Fancier graphics with ggplot</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">reshape2</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)

<span class="no">df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/tidyeval.html">exprs</a></span>(<span class="no">eset</span>)))
<span class="no">df</span>$<span class="no">sample</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html">sampleNames</a></span>(<span class="no">eset</span>)
<span class="no">df</span>$<span class="no">diseasestate</span> <span class="kw">&lt;-</span> <span class="no">eset</span>$<span class="no">Characteristics.DiseaseState</span>
<span class="no">df.melt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span>(<span class="no">df</span>, <span class="kw">id</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"sample"</span>,<span class="st">"diseasestate"</span>))

<span class="no">p1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">df.melt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">sample</span>), <span class="no">value</span>))+
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">fill</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">diseasestate</span>)), <span class="kw">outlier.shape</span> <span class="kw">=</span> <span class="fl">95</span>, <span class="kw">outlier.size</span> <span class="kw">=</span> <span class="fl">1</span>)+
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">"sample"</span>)+<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"expression"</span>)+
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()+
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">axis.text.x</span>  <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>()) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Boxplot of expression values by sample"</span>)
<span class="no">p1</span></pre></body></html></div>
<p><img src="microarrayQC_files/figure-html/boxplot-3.png" width="700"></p>
</div>
<div id="testing-the-log-normality-of-the-gene-distributions" class="section level2">
<h2 class="hasAnchor">
<a href="#testing-the-log-normality-of-the-gene-distributions" class="anchor"></a>Testing the (log-)normality of the gene distributions</h2>
<p>A lot of the statistical analyses performed on microarray data are based on the assumption that the data is (log-)normally distributed. Below, we show the use of a Kolmogorov-Smirnov (KS) test to compare each gene’s (log-)distribution to a standard normal.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="co">## simple wrapper to have ks.test return the pair &lt;statistic,p-value&gt;</span>
<span class="no">ks.pair</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">X</span>,<span class="no">Y</span>) { <span class="no">tmp</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/ks.test.html">ks.test</a></span>(<span class="no">X</span>,<span class="no">Y</span>); <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">ks</span><span class="kw">=</span><span class="no">tmp</span>$<span class="no">statistic</span>,<span class="kw">p</span><span class="kw">=</span><span class="no">tmp</span>$<span class="no">p.value</span>) }

<span class="co">## now let's apply it to each gene in the dataset</span>
<span class="no">emat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/tidyeval.html">exprs</a></span>(<span class="no">eset</span>)
<span class="no">emat01</span> <span class="kw">&lt;-</span> (<span class="no">emat</span> - <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="no">emat</span>)) / <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">emat</span>,<span class="fl">1</span>,<span class="no">sd</span>)
<span class="no">KS</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">emat01</span>,<span class="fl">1</span>,<span class="no">ks.pair</span>,<span class="no">pnorm</span>))

<span class="co">## let's plot the deviation from normality (given by -log10(p)) as a function of the gene's expression</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="no">emat</span>),-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS</span>[,<span class="st">"p"</span>]),<span class="kw">ylab</span><span class="kw">=</span><span class="st">"-log10(p)"</span>,<span class="kw">xlab</span><span class="kw">=</span><span class="st">"mean(gene)"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="kw">h</span><span class="kw">=</span>-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>),<span class="kw">col</span><span class="kw">=</span><span class="st">"red"</span>)
<span class="no">idx</span> <span class="kw">&lt;-</span> -<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS</span>[,<span class="st">"p"</span>])<span class="kw">&gt;</span>-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="no">emat</span>)[<span class="no">idx</span>],-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS</span>[<span class="no">idx</span>,<span class="st">"p"</span>]),<span class="kw">col</span><span class="kw">=</span><span class="st">"red"</span>)</pre></body></html></div>
<p><img src="microarrayQC_files/figure-html/test.lognormal-1.png" width="700"></p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="co">## let's compute a more conservative mht-corrected FDR, and plot that</span>
<span class="no">KS</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="no">KS</span>,<span class="kw">fdr</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(<span class="no">KS</span>[,<span class="st">"p"</span>]))
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="no">emat</span>),-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS</span>[,<span class="st">"fdr"</span>]),<span class="kw">ylab</span><span class="kw">=</span><span class="st">"-log10(FDR)"</span>,<span class="kw">xlab</span><span class="kw">=</span><span class="st">"mean(gene)"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="kw">h</span><span class="kw">=</span>-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>),<span class="kw">col</span><span class="kw">=</span><span class="st">"red"</span>)
<span class="no">idx</span> <span class="kw">&lt;-</span> -<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS</span>[,<span class="st">"fdr"</span>])<span class="kw">&gt;</span>-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="no">emat</span>)[<span class="no">idx</span>],-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS</span>[<span class="no">idx</span>,<span class="st">"fdr"</span>]),<span class="kw">col</span><span class="kw">=</span><span class="st">"red"</span>)</pre></body></html></div>
<p><img src="microarrayQC_files/figure-html/test.lognormal-2.png" width="700"></p>
<p>Here, we show an alternative ‘implementation’ where, instead of standardizing the data (emat), we pass to <code>pnorm</code> the gene mean and sd.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="co">## Rather than transforming the data to emat01, we can use the original data and add</span>
<span class="co">## mean = mean(X), sd = sd(X) to the ks.test() function.</span>
<span class="co">## This will feed the mean and sd of the gene-wise expression to the pnorm function</span>
<span class="no">ks.pnorm</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">X</span>,<span class="no">Y</span>) { <span class="no">tmp</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/ks.test.html">ks.test</a></span>(<span class="no">X</span>,<span class="no">pnorm</span>, <span class="kw">mean</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(<span class="no">X</span>), <span class="kw">sd</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span>(<span class="no">X</span>)); <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">ks</span><span class="kw">=</span><span class="no">tmp</span>$<span class="no">statistic</span>,<span class="kw">p</span><span class="kw">=</span><span class="no">tmp</span>$<span class="no">p.value</span>) }

<span class="co">## now we can run as before, but with the emat object, not emat01</span>
<span class="no">KS1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">emat</span>,<span class="fl">1</span>,<span class="no">ks.pnorm</span>))

<span class="co">## let's plot the deviation from normality (given by -log10(p)) as a function of the gene's expression</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="no">emat</span>),-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS1</span>[,<span class="st">"p"</span>]),<span class="kw">ylab</span><span class="kw">=</span><span class="st">"-log10(p)"</span>,<span class="kw">xlab</span><span class="kw">=</span><span class="st">"mean(gene)"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="kw">h</span><span class="kw">=</span>-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>),<span class="kw">col</span><span class="kw">=</span><span class="st">"red"</span>)
<span class="no">idx</span> <span class="kw">&lt;-</span> -<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS1</span>[,<span class="st">"p"</span>])<span class="kw">&gt;</span>-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="no">emat</span>)[<span class="no">idx</span>],-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS1</span>[<span class="no">idx</span>,<span class="st">"p"</span>]),<span class="kw">col</span><span class="kw">=</span><span class="st">"red"</span>)</pre></body></html></div>
<p><img src="microarrayQC_files/figure-html/test.lognormalAlt-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="co">## let's compute a more conservative mht-corrected FDR, and plot that</span>
<span class="no">KS1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="no">KS1</span>,<span class="kw">fdr</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(<span class="no">KS</span>[,<span class="st">"p"</span>]))
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="no">emat</span>),-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS1</span>[,<span class="st">"fdr"</span>]),<span class="kw">ylab</span><span class="kw">=</span><span class="st">"-log10(FDR)"</span>,<span class="kw">xlab</span><span class="kw">=</span><span class="st">"mean(gene)"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="kw">h</span><span class="kw">=</span>-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>),<span class="kw">col</span><span class="kw">=</span><span class="st">"red"</span>)
<span class="no">idx</span> <span class="kw">&lt;-</span> -<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS1</span>[,<span class="st">"fdr"</span>])<span class="kw">&gt;</span>-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="no">emat</span>)[<span class="no">idx</span>],-<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">KS1</span>[<span class="no">idx</span>,<span class="st">"fdr"</span>]),<span class="kw">col</span><span class="kw">=</span><span class="st">"red"</span>)</pre></body></html></div>
<p><img src="microarrayQC_files/figure-html/test.lognormalAlt-2.png" width="700"></p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="co"># Compare the test statistics</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="no">KS</span>[,<span class="st">"ks.D"</span>], <span class="no">KS1</span>[,<span class="st">"ks.D"</span>])</pre></body></html></div>
<p><img src="microarrayQC_files/figure-html/test.lognormalAlt-3.png" width="700"></p>
</div>
<div id="variation-filtering" class="section level2">
<h2 class="hasAnchor">
<a href="#variation-filtering" class="anchor"></a>Variation Filtering</h2>
<p>We now show how to filter genes based on different measures of variation: standrad deviation (SD) and its robust counterpart, the median absolute deviation (MAD).</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="co">## variation filter: rank genes according to that variation across samples</span>
<span class="co">## method 1: filter by standard deviation</span>
<span class="no">MEAN</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">emat</span>, <span class="fl">1</span>, <span class="no">mean</span>)
<span class="no">STDEV</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">emat</span>, <span class="fl">1</span>, <span class="no">sd</span>)
<span class="no">MAD</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">emat</span>, <span class="fl">1</span>, <span class="no">mad</span>)
<span class="no">MEDIAN</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">emat</span>, <span class="fl">1</span>, <span class="no">median</span>)

<span class="co">## highlight the top 4000 genes by stdev</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="no">MEAN</span>, <span class="no">STDEV</span>, <span class="kw">pch</span> <span class="kw">=</span> <span class="st">"."</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"mean vs. stdev (log2 data)"</span>)
<span class="no">top4Ksd</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="no">STDEV</span>, <span class="kw">decreasing</span> <span class="kw">=</span> <span class="no">T</span>)[<span class="fl">1</span>:<span class="fl">4000</span>]
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="no">MEAN</span>[<span class="no">top4Ksd</span>], <span class="no">STDEV</span>[<span class="no">top4Ksd</span>], <span class="kw">pch</span> <span class="kw">=</span> <span class="st">"."</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="st">"red"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"topright"</span>, <span class="kw">pch</span> <span class="kw">=</span> <span class="fl">20</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="st">"red"</span>, <span class="kw">legend</span> <span class="kw">=</span> <span class="st">"top 4K genes by stdev"</span>)</pre></body></html></div>
<p><img src="microarrayQC_files/figure-html/variationfilter-1.png" width="700"></p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">eset.top4ksd</span> <span class="kw">&lt;-</span> <span class="no">eset</span>[<span class="no">top4Ksd</span>,]
<span class="co">#saveRDS(eset.top4ksd, file=file.path(OMPATH,"data/AEDAT.collapsed.sd4k.RDS"))</span>

<span class="co">## unlogged data (multiplicative error), but plotted on logged axis</span>
<span class="no">MEANUNLOGGED</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="fl">2</span>^<span class="no">emat</span>)
<span class="no">SDUNLOGGED</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="fl">2</span>^<span class="no">emat</span>,<span class="fl">1</span>,<span class="no">sd</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="no">MEANUNLOGGED</span>,<span class="no">SDUNLOGGED</span>,<span class="kw">pch</span><span class="kw">=</span><span class="st">"."</span>,<span class="kw">log</span><span class="kw">=</span><span class="st">"xy"</span>,<span class="kw">main</span><span class="kw">=</span><span class="st">"mean vs. stdev (unlogged data)"</span>)
<span class="no">top4K</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="no">SDUNLOGGED</span>,<span class="kw">decreasing</span><span class="kw">=</span><span class="no">T</span>)[<span class="fl">1</span>:<span class="fl">4000</span>]
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="no">MEANUNLOGGED</span>[<span class="no">top4K</span>],<span class="no">SDUNLOGGED</span>[<span class="no">top4K</span>],<span class="kw">pch</span><span class="kw">=</span><span class="st">"."</span>,<span class="kw">col</span><span class="kw">=</span><span class="st">"red"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"topright"</span>,<span class="kw">pch</span><span class="kw">=</span><span class="fl">20</span>,<span class="kw">col</span><span class="kw">=</span><span class="st">"red"</span>,<span class="kw">legend</span><span class="kw">=</span><span class="st">"top 4K genes by stdev(unlogged)"</span>)</pre></body></html></div>
<p><img src="microarrayQC_files/figure-html/variationfilter-2.png" width="700"></p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="co">## method 2: filter by MAD (median absolute deviation)</span>
<span class="no">MAD</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">emat</span>, <span class="fl">1</span>, <span class="no">mad</span>)
<span class="no">MEDIAN</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">emat</span>, <span class="fl">1</span>, <span class="no">median</span>)

<span class="co">## highlight top 4000 genes by MAD</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="no">MEDIAN</span>, <span class="no">MAD</span>, <span class="kw">pch</span> <span class="kw">=</span> <span class="st">"."</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"median vs. MAD (log2 data)"</span>)
<span class="no">top4Kmad</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="no">MAD</span>, <span class="kw">decreasing</span> <span class="kw">=</span> <span class="no">T</span>)[<span class="fl">1</span>:<span class="fl">4000</span>]
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="no">MEDIAN</span>[<span class="no">top4Kmad</span>], <span class="no">MAD</span>[<span class="no">top4Kmad</span>], <span class="kw">pch</span> <span class="kw">=</span> <span class="st">"."</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="st">"red"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"topright"</span>, <span class="kw">pch</span> <span class="kw">=</span> <span class="fl">20</span>, <span class="kw">col</span> <span class="kw">=</span> <span class="st">"red"</span>, <span class="kw">legend</span> <span class="kw">=</span> <span class="st">"top 4K genes by MAD"</span>)</pre></body></html></div>
<p><img src="microarrayQC_files/figure-html/variationfilter-3.png" width="700"></p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="no">eset.top4kmad</span> <span class="kw">&lt;-</span> <span class="no">eset</span>[<span class="no">top4Kmad</span>,]
<span class="co">#saveRDS(eset.top4kmad, file=file.path(OMPATH,"data/AEDAT.collapsed.mad4k.RDS"))</span>

<span class="co">## let us look at the overlap between the two filters</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span>(<span class="no">top4Ksd</span>,<span class="no">top4Kmad</span>))</pre></body></html></div>
<pre><code>## [1] 3058</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/tetomonti">Stefano Monti</a>, <a href="https://github.com/anfederico">Anthony Federico</a>, <a href="https://github.com/ericreed">Eric Reed</a>, <a href="https://github.com/lia978">Amy Li</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.0.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
