<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Quality Control • BS831</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Quality Control">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">BS831</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Markdowns
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Introduction to Genomics Analysis</li>
    <li>
      <a href="../../articles/docs/matrixOp.html">Efficient Computation in R</a>
    </li>
    <li>
      <a href="../../articles/docs/ExpressionSet.html">The ExpressionSet Data Object</a>
    </li>
    <li>
      <a href="../../articles/docs/Distributions.html">Probability Distributions</a>
    </li>
    <li>
      <a href="../../articles/docs/Mix_Normals.html">Mixture of Normals</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Preprocessing and Quality Control</li>
    <li>
      <a href="../../articles/docs/Microarray_Processing.html">Data Acquisition and ExpressionSet Basics</a>
    </li>
    <li>
      <a href="../../articles/docs/microarrayQC.html">Quality Control</a>
    </li>
    <li>
      <a href="../../articles/docs/Dimensionality_Reduction.html">Dimensionality Reduction</a>
    </li>
    <li>
      <a href="../../articles/docs/Heatmaps.html">Heatmaps</a>
    </li>
    <li>
      <a href="../../articles/docs/hclust.html">Hierarchical Clustering</a>
    </li>
    <li>
      <a href="../../articles/docs/quantileNormalization.html">Quantile Normalization</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments I: Microarrays and Linear Models</li>
    <li>
      <a href="../../articles/docs/SampleSize.html">Sample Size and Statistical Significance</a>
    </li>
    <li>
      <a href="../../articles/docs/MHTcorrection.html">Multiple-Hypothesis Correction</a>
    </li>
    <li>
      <a href="../../articles/docs/Linear_Regression.html">Linear Regression</a>
    </li>
    <li>
      <a href="../../articles/docs/diffanalLM.html">Differential Analysis as Linear Regression</a>
    </li>
    <li>
      <a href="../../articles/docs/Diffanalysis.html">Gene Expression Differential Analysis with Microarrays</a>
    </li>
    <li>
      <a href="../../articles/docs/Diffanal_LimmaBC.html">Gene Expression Differential Analysis based on Limma</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments II: RNA-seq and Generalized Lineary Models</li>
    <li>
      <a href="../../articles/docs/RNAseq_ScalNorm.html">RNA-Seq Data Scaling and Normalization</a>
    </li>
    <li>
      <a href="../../articles/docs/DiffanalysisRNAseqComparison.html">RNA-seq Differential Expression Analysis with DEseq2, edgeR and limma</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Comparative Experiments III: Differential Enrichment Analysis</li>
    <li>
      <a href="../../articles/docs/HyperEnrichment.html">HyperGeometric Enrichment</a>
    </li>
    <li>
      <a href="../../articles/docs/KS_Enrichment.html">KS-based Enrichment</a>
    </li>
    <li>
      <a href="../../articles/docs/GeneSetProjectionGSVA.html">Gene Set Projection based on KS</a>
    </li>
    <li>
      <a href="../../articles/docs/GeneSetProjectionASSIGN.html">Geneset projection based on ASSIGN</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Classification</li>
    <li>
      <a href="../../articles/docs/featureRedundancy.html">Feature Redundancy: the Good and the Bad</a>
    </li>
    <li>
      <a href="../../articles/docs/Classification_Caret.html">Classification Using Caret</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Clustering</li>
    <li>
      <a href="../../articles/docs/GenesetMapping4KS.html">Mapping genesets for KS testing</a>
    </li>
    <li>
      <a href="../../articles/docs/Clustering.html">Clustering and Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/montilab/BS831">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Quality Control</h1>
                        <h4 class="author">Stefano Monti</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/montilab/BS831/blob/master/vignettes/docs/microarrayQC.Rmd"><code>vignettes/docs/microarrayQC.Rmd</code></a></small>
      <div class="hidden name"><code>microarrayQC.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BS831)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(Biobase)</span></code></pre></div>
<p>Here we show some basic QC plots to assess the quality of the data and the nature of the distribution. For this purpose, we use a small Breast Cancer dataset we have pre-uploaded to the <code>data/</code> folder.</p>
<div id="import-breast-cancer-data" class="section level2">
<h2 class="hasAnchor">
<a href="#import-breast-cancer-data" class="anchor"></a>Import Breast Cancer Data</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co">## breast cancer subtypes dataset from ArrayExpress</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(AEDAT.collapsed)</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a>eset &lt;-<span class="st"> </span>AEDAT.collapsed</span>
<span id="cb2-5"><a href="#cb2-5"></a>emat &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(eset)</span>
<span id="cb2-6"><a href="#cb2-6"></a></span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="co">## Assign rownames to emat that are gene annotations (These were removed by mistake)</span></span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(emat) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html">fData</a></span>(eset))</span>
<span id="cb2-9"><a href="#cb2-9"></a></span>
<span id="cb2-10"><a href="#cb2-10"></a><span class="co">## Table of disease states</span></span>
<span id="cb2-11"><a href="#cb2-11"></a><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/table.html">table</a></span>(eset<span class="op">$</span>Characteristics.DiseaseState))</span></code></pre></div>
<pre><code>## 
##    BRCA1-associated breast cancer      non-basal-like breast cancer 
##                                 2                                20 
##                            normal sporadic basal-like breast cancer 
##                                 7                                18</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co">## simplify labels</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>eset<span class="op">$</span>Characteristics.DiseaseState &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">" breast cancer"</span>,<span class="st">""</span>,eset<span class="op">$</span>Characteristics.DiseaseState)</span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/table.html">table</a></span>(eset<span class="op">$</span>Characteristics.DiseaseState))</span></code></pre></div>
<pre><code>## 
##    BRCA1-associated      non-basal-like              normal sporadic basal-like 
##                   2                  20                   7                  18</code></pre>
</div>
<div id="boxplots-of-gene-expression-values-by-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#boxplots-of-gene-expression-values-by-sample" class="anchor"></a>Boxplots of gene expression values by sample</h2>
<p>Here, we plot the distribution of genes within each sample as boxplots. This is a quick check of how symmetric the distributions are, and whether any ‘biases’ are apparent.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="co">## Boxplot of expression values per sample (sorted and colored by disease state)</span></span>
<span id="cb6-2"><a href="#cb6-2"></a>sampleCol &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"coral"</span>,<span class="st">"green"</span>,<span class="st">"blue"</span>,<span class="st">"purple"</span>)[<span class="kw"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(eset<span class="op">$</span>Characteristics.DiseaseState)]</span>
<span id="cb6-3"><a href="#cb6-3"></a>ord &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/order.html">order</a></span>( eset<span class="op">$</span>Characteristics.DiseaseState )</span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="kw"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(eset)[,ord]),<span class="dt">col=</span>sampleCol[ord],<span class="dt">pch=</span><span class="st">"-"</span>)</span></code></pre></div>
<p><img src="microarrayQC_files/figure-html/boxplot-1.png" width="700"></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co">## If you have a slow machine, you might want to subsample (random 1000 genes from the total)</span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="co">## here, we are also taking the opportunity to better size (cex) and orient (las) the axis labels</span></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="kw"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="dv">321</span>) <span class="co"># for reproducible results</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="kw"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(eset)[<span class="kw"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(emat),<span class="dv">1000</span>),ord]),<span class="dt">col=</span>sampleCol[ord],<span class="dt">pch=</span><span class="st">"-"</span>,</span>
<span id="cb7-5"><a href="#cb7-5"></a>        <span class="dt">las=</span><span class="dv">2</span>,<span class="dt">cex.axis=</span>.<span class="dv">75</span>)</span></code></pre></div>
<p><img src="microarrayQC_files/figure-html/boxplot-2.png" width="700"></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co">## Fancier graphics with ggplot</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(reshape2)</span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ggplot2)</span>
<span id="cb8-4"><a href="#cb8-4"></a></span>
<span id="cb8-5"><a href="#cb8-5"></a>df &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/tidyeval.html">exprs</a></span>(eset)))</span>
<span id="cb8-6"><a href="#cb8-6"></a>df<span class="op">$</span>sample &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html">sampleNames</a></span>(eset)</span>
<span id="cb8-7"><a href="#cb8-7"></a>df<span class="op">$</span>diseasestate &lt;-<span class="st"> </span>eset<span class="op">$</span>Characteristics.DiseaseState</span>
<span id="cb8-8"><a href="#cb8-8"></a>df.melt &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span>(df, <span class="dt">id=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"sample"</span>,<span class="st">"diseasestate"</span>))</span>
<span id="cb8-9"><a href="#cb8-9"></a></span>
<span id="cb8-10"><a href="#cb8-10"></a>p1 &lt;-<span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(df.melt, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(sample), value))<span class="op">+</span></span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">fill =</span> <span class="kw"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(diseasestate)), <span class="dt">outlier.shape =</span> <span class="dv">95</span>, <span class="dt">outlier.size =</span> <span class="dv">1</span>)<span class="op">+</span></span>
<span id="cb8-12"><a href="#cb8-12"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">"sample"</span>)<span class="op">+</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"expression"</span>)<span class="op">+</span></span>
<span id="cb8-13"><a href="#cb8-13"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()<span class="op">+</span></span>
<span id="cb8-14"><a href="#cb8-14"></a><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">axis.text.x  =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>()) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Boxplot of expression values by sample"</span>)</span>
<span id="cb8-15"><a href="#cb8-15"></a>p1</span></code></pre></div>
<p><img src="microarrayQC_files/figure-html/boxplot-3.png" width="700"></p>
</div>
<div id="testing-the-log-normality-of-the-gene-distributions" class="section level2">
<h2 class="hasAnchor">
<a href="#testing-the-log-normality-of-the-gene-distributions" class="anchor"></a>Testing the (log-)normality of the gene distributions</h2>
<p>A lot of the statistical analyses performed on microarray data are based on the assumption that the data is (log-)normally distributed. Below, we show the use of a Kolmogorov-Smirnov (KS) test to compare each gene’s (log-)distribution to a standard normal.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="co">## simple wrapper to have ks.test return the pair &lt;statistic,p-value&gt;</span></span>
<span id="cb9-2"><a href="#cb9-2"></a>ks.pair &lt;-<span class="st"> </span><span class="cf">function</span>(X,Y) { tmp &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/ks.test.html">ks.test</a></span>(X,Y); <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">ks=</span>tmp<span class="op">$</span>statistic,<span class="dt">p=</span>tmp<span class="op">$</span>p.value) }</span>
<span id="cb9-3"><a href="#cb9-3"></a></span>
<span id="cb9-4"><a href="#cb9-4"></a><span class="co">## now let's apply it to each gene in the dataset</span></span>
<span id="cb9-5"><a href="#cb9-5"></a>emat &lt;-<span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/tidyeval.html">exprs</a></span>(eset)</span>
<span id="cb9-6"><a href="#cb9-6"></a>emat01 &lt;-<span class="st"> </span>(emat <span class="op">-</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(emat)) <span class="op">/</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(emat,<span class="dv">1</span>,sd)</span>
<span id="cb9-7"><a href="#cb9-7"></a>KS &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(emat01,<span class="dv">1</span>,ks.pair,pnorm))</span>
<span id="cb9-8"><a href="#cb9-8"></a></span>
<span id="cb9-9"><a href="#cb9-9"></a><span class="co">## let's plot the deviation from normality (given by -log10(p)) as a function of the gene's expression</span></span>
<span id="cb9-10"><a href="#cb9-10"></a><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(emat),<span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS[,<span class="st">"p"</span>]),<span class="dt">ylab=</span><span class="st">"-log10(p)"</span>,<span class="dt">xlab=</span><span class="st">"mean(gene)"</span>)</span>
<span id="cb9-11"><a href="#cb9-11"></a><span class="kw"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="dt">h=</span><span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>),<span class="dt">col=</span><span class="st">"red"</span>)</span>
<span id="cb9-12"><a href="#cb9-12"></a>idx &lt;-<span class="st"> </span><span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS[,<span class="st">"p"</span>])<span class="op">&gt;-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>)</span>
<span id="cb9-13"><a href="#cb9-13"></a><span class="kw"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(emat)[idx],<span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS[idx,<span class="st">"p"</span>]),<span class="dt">col=</span><span class="st">"red"</span>)</span></code></pre></div>
<p><img src="microarrayQC_files/figure-html/test.lognormal-1.png" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co">## let's compute a more conservative mht-corrected FDR, and plot that</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>KS &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(KS,<span class="dt">fdr=</span><span class="kw"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(KS[,<span class="st">"p"</span>]))</span>
<span id="cb10-3"><a href="#cb10-3"></a><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(emat),<span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS[,<span class="st">"fdr"</span>]),<span class="dt">ylab=</span><span class="st">"-log10(FDR)"</span>,<span class="dt">xlab=</span><span class="st">"mean(gene)"</span>)</span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="kw"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="dt">h=</span><span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>),<span class="dt">col=</span><span class="st">"red"</span>)</span>
<span id="cb10-5"><a href="#cb10-5"></a>idx &lt;-<span class="st"> </span><span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS[,<span class="st">"fdr"</span>])<span class="op">&gt;-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>)</span>
<span id="cb10-6"><a href="#cb10-6"></a><span class="kw"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(emat)[idx],<span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS[idx,<span class="st">"fdr"</span>]),<span class="dt">col=</span><span class="st">"red"</span>)</span></code></pre></div>
<p><img src="microarrayQC_files/figure-html/test.lognormal-2.png" width="700"></p>
<p>Here, we show an alternative ‘implementation’ where, instead of standardizing the data (emat), we pass to <code>pnorm</code> the gene mean and sd.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co">## Rather than transforming the data to emat01, we can use the original data and add</span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="co">## mean = mean(X), sd = sd(X) to the ks.test() function.</span></span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="co">## This will feed the mean and sd of the gene-wise expression to the pnorm function</span></span>
<span id="cb11-4"><a href="#cb11-4"></a>ks.pnorm &lt;-<span class="st"> </span><span class="cf">function</span>(X,Y) { tmp &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/ks.test.html">ks.test</a></span>(X,pnorm, <span class="dt">mean=</span><span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(X), <span class="dt">sd=</span><span class="kw"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span>(X)); <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">ks=</span>tmp<span class="op">$</span>statistic,<span class="dt">p=</span>tmp<span class="op">$</span>p.value) }</span>
<span id="cb11-5"><a href="#cb11-5"></a></span>
<span id="cb11-6"><a href="#cb11-6"></a><span class="co">## now we can run as before, but with the emat object, not emat01</span></span>
<span id="cb11-7"><a href="#cb11-7"></a>KS1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(emat,<span class="dv">1</span>,ks.pnorm))</span>
<span id="cb11-8"><a href="#cb11-8"></a></span>
<span id="cb11-9"><a href="#cb11-9"></a><span class="co">## let's plot the deviation from normality (given by -log10(p)) as a function of the gene's expression</span></span>
<span id="cb11-10"><a href="#cb11-10"></a><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(emat),<span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS1[,<span class="st">"p"</span>]),<span class="dt">ylab=</span><span class="st">"-log10(p)"</span>,<span class="dt">xlab=</span><span class="st">"mean(gene)"</span>)</span>
<span id="cb11-11"><a href="#cb11-11"></a><span class="kw"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="dt">h=</span><span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>),<span class="dt">col=</span><span class="st">"red"</span>)</span>
<span id="cb11-12"><a href="#cb11-12"></a>idx &lt;-<span class="st"> </span><span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS1[,<span class="st">"p"</span>])<span class="op">&gt;-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>)</span>
<span id="cb11-13"><a href="#cb11-13"></a><span class="kw"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(emat)[idx],<span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS1[idx,<span class="st">"p"</span>]),<span class="dt">col=</span><span class="st">"red"</span>)</span></code></pre></div>
<p><img src="microarrayQC_files/figure-html/test.lognormalAlt-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="co">## let's compute a more conservative mht-corrected FDR, and plot that</span></span>
<span id="cb12-2"><a href="#cb12-2"></a>KS1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(KS1,<span class="dt">fdr=</span><span class="kw"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span>(KS[,<span class="st">"p"</span>]))</span>
<span id="cb12-3"><a href="#cb12-3"></a><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(emat),<span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS1[,<span class="st">"fdr"</span>]),<span class="dt">ylab=</span><span class="st">"-log10(FDR)"</span>,<span class="dt">xlab=</span><span class="st">"mean(gene)"</span>)</span>
<span id="cb12-4"><a href="#cb12-4"></a><span class="kw"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="dt">h=</span><span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>),<span class="dt">col=</span><span class="st">"red"</span>)</span>
<span id="cb12-5"><a href="#cb12-5"></a>idx &lt;-<span class="st"> </span><span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS1[,<span class="st">"fdr"</span>])<span class="op">&gt;-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.01</span>)</span>
<span id="cb12-6"><a href="#cb12-6"></a><span class="kw"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(emat)[idx],<span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(KS1[idx,<span class="st">"fdr"</span>]),<span class="dt">col=</span><span class="st">"red"</span>)</span></code></pre></div>
<p><img src="microarrayQC_files/figure-html/test.lognormalAlt-2.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co"># Compare the test statistics</span></span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(KS[,<span class="st">"ks.D"</span>], KS1[,<span class="st">"ks.D"</span>])</span></code></pre></div>
<p><img src="microarrayQC_files/figure-html/test.lognormalAlt-3.png" width="700"></p>
</div>
<div id="variation-filtering" class="section level2">
<h2 class="hasAnchor">
<a href="#variation-filtering" class="anchor"></a>Variation Filtering</h2>
<p>We now show how to filter genes based on different measures of variation: standrad deviation (SD) and its robust counterpart, the median absolute deviation (MAD).</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="co">## variation filter: rank genes according to that variation across samples</span></span>
<span id="cb14-2"><a href="#cb14-2"></a><span class="co">## method 1: filter by standard deviation</span></span>
<span id="cb14-3"><a href="#cb14-3"></a>MEAN &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(emat, <span class="dv">1</span>, mean)</span>
<span id="cb14-4"><a href="#cb14-4"></a>STDEV &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(emat, <span class="dv">1</span>, sd)</span>
<span id="cb14-5"><a href="#cb14-5"></a>MAD &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(emat, <span class="dv">1</span>, mad)</span>
<span id="cb14-6"><a href="#cb14-6"></a>MEDIAN &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(emat, <span class="dv">1</span>, median)</span>
<span id="cb14-7"><a href="#cb14-7"></a></span>
<span id="cb14-8"><a href="#cb14-8"></a><span class="co">## highlight the top 4000 genes by stdev</span></span>
<span id="cb14-9"><a href="#cb14-9"></a><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(MEAN, STDEV, <span class="dt">pch =</span> <span class="st">"."</span>, <span class="dt">main =</span> <span class="st">"mean vs. stdev (log2 data)"</span>)</span>
<span id="cb14-10"><a href="#cb14-10"></a>top4Ksd &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/order.html">order</a></span>(STDEV, <span class="dt">decreasing =</span> T)[<span class="dv">1</span><span class="op">:</span><span class="dv">4000</span>]</span>
<span id="cb14-11"><a href="#cb14-11"></a><span class="kw"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(MEAN[top4Ksd], STDEV[top4Ksd], <span class="dt">pch =</span> <span class="st">"."</span>, <span class="dt">col =</span> <span class="st">"red"</span>)</span>
<span id="cb14-12"><a href="#cb14-12"></a><span class="kw"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"topright"</span>, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">legend =</span> <span class="st">"top 4K genes by stdev"</span>)</span></code></pre></div>
<p><img src="microarrayQC_files/figure-html/variationfilter-1.png" width="700"></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>eset.top4ksd &lt;-<span class="st"> </span>eset[top4Ksd,]</span>
<span id="cb15-2"><a href="#cb15-2"></a><span class="co">#saveRDS(eset.top4ksd, file=file.path(OMPATH,"data/AEDAT.collapsed.sd4k.RDS"))</span></span>
<span id="cb15-3"><a href="#cb15-3"></a></span>
<span id="cb15-4"><a href="#cb15-4"></a><span class="co">## unlogged data (multiplicative error), but plotted on logged axis</span></span>
<span id="cb15-5"><a href="#cb15-5"></a>MEANUNLOGGED &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span>(<span class="dv">2</span><span class="op">^</span>emat)</span>
<span id="cb15-6"><a href="#cb15-6"></a>SDUNLOGGED &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="dv">2</span><span class="op">^</span>emat,<span class="dv">1</span>,sd)</span>
<span id="cb15-7"><a href="#cb15-7"></a><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(MEANUNLOGGED,SDUNLOGGED,<span class="dt">pch=</span><span class="st">"."</span>,<span class="dt">log=</span><span class="st">"xy"</span>,<span class="dt">main=</span><span class="st">"mean vs. stdev (unlogged data)"</span>)</span>
<span id="cb15-8"><a href="#cb15-8"></a>top4K &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/order.html">order</a></span>(SDUNLOGGED,<span class="dt">decreasing=</span>T)[<span class="dv">1</span><span class="op">:</span><span class="dv">4000</span>]</span>
<span id="cb15-9"><a href="#cb15-9"></a><span class="kw"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(MEANUNLOGGED[top4K],SDUNLOGGED[top4K],<span class="dt">pch=</span><span class="st">"."</span>,<span class="dt">col=</span><span class="st">"red"</span>)</span>
<span id="cb15-10"><a href="#cb15-10"></a><span class="kw"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"topright"</span>,<span class="dt">pch=</span><span class="dv">20</span>,<span class="dt">col=</span><span class="st">"red"</span>,<span class="dt">legend=</span><span class="st">"top 4K genes by stdev(unlogged)"</span>)</span></code></pre></div>
<p><img src="microarrayQC_files/figure-html/variationfilter-2.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="co">## method 2: filter by MAD (median absolute deviation)</span></span>
<span id="cb16-2"><a href="#cb16-2"></a>MAD &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(emat, <span class="dv">1</span>, mad)</span>
<span id="cb16-3"><a href="#cb16-3"></a>MEDIAN &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(emat, <span class="dv">1</span>, median)</span>
<span id="cb16-4"><a href="#cb16-4"></a></span>
<span id="cb16-5"><a href="#cb16-5"></a><span class="co">## highlight top 4000 genes by MAD</span></span>
<span id="cb16-6"><a href="#cb16-6"></a><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(MEDIAN, MAD, <span class="dt">pch =</span> <span class="st">"."</span>, <span class="dt">main =</span> <span class="st">"median vs. MAD (log2 data)"</span>)</span>
<span id="cb16-7"><a href="#cb16-7"></a>top4Kmad &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/order.html">order</a></span>(MAD, <span class="dt">decreasing =</span> T)[<span class="dv">1</span><span class="op">:</span><span class="dv">4000</span>]</span>
<span id="cb16-8"><a href="#cb16-8"></a><span class="kw"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(MEDIAN[top4Kmad], MAD[top4Kmad], <span class="dt">pch =</span> <span class="st">"."</span>, <span class="dt">col =</span> <span class="st">"red"</span>)</span>
<span id="cb16-9"><a href="#cb16-9"></a><span class="kw"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"topright"</span>, <span class="dt">pch =</span> <span class="dv">20</span>, <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">legend =</span> <span class="st">"top 4K genes by MAD"</span>)</span></code></pre></div>
<p><img src="microarrayQC_files/figure-html/variationfilter-3.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>eset.top4kmad &lt;-<span class="st"> </span>eset[top4Kmad,]</span>
<span id="cb17-2"><a href="#cb17-2"></a><span class="co">#saveRDS(eset.top4kmad, file=file.path(OMPATH,"data/AEDAT.collapsed.mad4k.RDS"))</span></span>
<span id="cb17-3"><a href="#cb17-3"></a></span>
<span id="cb17-4"><a href="#cb17-4"></a><span class="co">## let us look at the overlap between the two filters</span></span>
<span id="cb17-5"><a href="#cb17-5"></a><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span>(top4Ksd,top4Kmad))</span></code></pre></div>
<pre><code>## [1] 3058</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#import-breast-cancer-data">Import Breast Cancer Data</a></li>
      <li><a href="#boxplots-of-gene-expression-values-by-sample">Boxplots of gene expression values by sample</a></li>
      <li><a href="#testing-the-log-normality-of-the-gene-distributions">Testing the (log-)normality of the gene distributions</a></li>
      <li><a href="#variation-filtering">Variation Filtering</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/tetomonti">Stefano Monti</a>, <a href="https://github.com/lia978">Amy Li</a>, <a href="https://github.com/ericreed">Eric Reed</a>, <a href="https://github.com/anfederico">Anthony Federico</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
